// spang-web/src/views/safety/task/component/risk.js
export const safetyForm = {
    riskType: [{
        type: '火灾事故',
        fillIn: ''
      },
      {
        type: '中毒或窒息',
        fillIn: ''
      },
      {
        type: '触电事故',
        fillIn: ''
      },
      {
        type: '车辆伤害',
        fillIn: ''
      },
      {
        type: '起重伤害',
        fillIn: ''
      },
      {
        type: '淹溺事故',
        fillIn: ''
      },
      {
        type: '物体打击',
        fillIn: ''
      },
      {
        type: '爆炸事故',
        fillIn: ''
      },
      {
        type: '伤害坠落',
        fillIn: ''
      },
      {
        type: '机械伤害',
        fillIn: ''
      },
      {
        type: '灼烫事故',
        fillIn: ''
      },
      {
        type: '坍塌事故',
        fillIn: ''
      },
      {
        type: '其它',
        fillIn: ''
      }
    ],
    measures: [{
        content: '特殊作业前办理作业票，并持票作业。'
      },
      {
        content: '清除动火地点周围易燃物。'
      },
      {
        content: '加盲板隔离或与有毒、易燃介质管线断开。'
      },
      {
        content: '测氧测爆有毒气体分析合格。'
      },
      {
        content: '通氮气或蒸汽进行保护。'
      },
      {
        content: '分析与动火间隔不得超过30分钟，如超过或动火间断超过30分钟，或作业条件发生变化，应重新分析。'
      },
      {
        content: '受限空间每两小时分析一次。'
      },
      {
        content: '分析要有代表性，构架或大型容器应在上中下分别采样分析。'
      },
      {
        content: '动火点15米内不得有可燃液体排放，30米内不得有可燃气体排放。'
      },
      {
        content: '专人监护，并不得离岗，发现异常立即停止作业，联系有关人员采取措施。'
      },
      {
        content: '加强通风。'
      },
      {
        content: '配备灭火器。'
      },
      {
        content: '地沟地井用防火毯覆盖。'
      },
      {
        content: '附近阀门填料、仪表箱用防火毯覆盖。'
      },
      {
        content: '拉警戒线，夜间配警示灯。'
      },
      {
        content: '清除动火地点周围易燃物。'
      },
      {
        content: '违反安全作业规程或监护人离岗，作业立即停止。'
      },
      {
        content: '作业工具检查合格。'
      },
      {
        content: '做好防火花飞溅措施。'
      },
      {
        content: '按规定正确佩戴防护用品（安全带、安全绳、防尘口罩、护目镜、安全帽、工作服、耐酸碱手套、空气呼吸器、长管呼吸器、滤毒盒、防火服、防化服、面屏等）。'
      },
      {
        content: '电焊回路应搭接在焊件上，不得与其他设备搭接，严禁穿越下水道。'
      },
      {
        content: '氧气乙炔气瓶间距不小于5米，二者与动火地点之间距离不小于10米。'
      },
      {
        content: '使用防爆低压灯具（干燥容器内36V，潮湿或狭小容器内12V）和防爆工具。'
      },
      {
        content: '气瓶不得在烈日下暴晒，乙炔瓶带有阻火器，禁止卧放。'
      },
      {
        content: '断电。'
      },
      {
        content: '电线不得有裸露接头，电缆和气带均不得有缠绕盘卷。'
      },
      {
        content: '患有职业禁忌症患者严禁进行高处作业。'
      },
      {
        content: '作业时感觉不适人员立即撤离。'
      },
      {
        content: '高处作业系好安全带，严禁抛掷工具、物品。'
      },
      {
        content: '吊拉物品时，要捆绑牢固，下面不得站人。'
      },
      {
        content: '作业结束时，做好现场清理及确认。'
      },
      {
        content: '其他安全措施、作业方案要求落实的补充内容或事故案例警示内容：',
        fillIn: ''
      }
    ]
  }
