// spang-web/src/views/safety/task/electric/risks.js
export const hotWorkRiskFillinList = [
    {
      titleId: 1,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '安装临时线路人员持有电工作业操作证',
      measureLabel: 'false'
    },
    {
      titleId: 2,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '在防爆场所使用的临时用电设备和线路达到相应的防爆等级要求或已办理动火证',
      measureLabel: '动火证编号',
      measuresValue: '',
      measuresText: ''
    },
    {
      titleId: 3,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      measuresValue: '',
      measuresText: '',
      label: '临时用电采用芯$mm2电缆，符合用电负荷要求。',
      measureLabel: 'false'
    },
    {
      titleId: 4,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '临时用电的单相和两相用电采用三芯电缆',
      measureLabel: 'false'
    },
    {
      titleId: 5,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '临时用电的三相用电采用四芯电缆',
      measureLabel: 'false'
    },
    {
      titleId: 6,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '临时用电的单相和三相混用线路采用五线制',
      measureLabel: 'false'
    },
    {
      titleId: 7,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '临时用电设施装有漏电保护器，移动工具、手持工具"一机一闸一保护"',
      showMeasureLabel: true,
      measureLabel: 'false'
    },
    {
      titleId: 8,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '临时用电设备接地良好、各接线处接触良好',
      measureLabel: 'false'
    },
    {
      titleId: 9,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      measuresValue: [],
      measuresText: '',
      label: '现场临时用电配电盘、配电箱柜门、锁具完好，有防雨措施，接地良好',
      measureLabel: 'false'
    },
    {
      titleId: 10,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '现场临时用电配电盘、配电箱接线良好整齐，箱内无其他可燃、易燃杂物',
      measureLabel: 'false'
    },
    {
      titleId: 11,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '暗管埋设及地下电缆线路设有"走向标志"和"安全标志", 电缆埋深大于0.7 m',
      measureLabel: 'false'
    },
    {
      titleId: 12,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '临时用电线路不得采用裸线，且不得捆绑在管线、阀门杆、脚手架、护栏等上',
      measureLabel: 'false'
    },
    {
      titleId: 13,
      isTitle: 1,
      titleValue: '',
      isMeasures: 0,
      measuresText: '',
      label: '其他措施',
      measureLabel: 'false'
    }
  ]
