// spang-web/src/views/safety/fire/columns.js
export const fireEquipmentColumns = [
  {
    title: '序号',
    dataIndex: 'number',
    width: '60px',
    align: 'center',
    scopedSlots: { customRender: 'number' }
  },
  {
    title: '编号',
    dataIndex: 'number',
    width: '120px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '消防器材名称',
    dataIndex: 'name',
    width: '150px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '规格型号',
    dataIndex: 'models',
    width: '150px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '位置',
    dataIndex: 'address',
    width: '180px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '责任人',
    dataIndex: 'chargePerson',
    width: '120px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '有效截止日期',
    dataIndex: 'expirationDate',
    width: '150px',
    align: 'center',
    customRender: (text, record, index) => `${text.substr(0,10)}`
  },
  {
    title: '状态',
    dataIndex: 'statusName',
    width: '120px',
    align: 'center',
    scopedSlots: { customRender: 'statusName' }
  },
  {
    title: '说明',
    dataIndex: 'explain',
    width: '100px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '180px',
    align: 'center',
    // fixed: 'right',
    scopedSlots: { customRender: 'action' }
  }
]

export const fireImportantPositionColumns = [
  {
    title: '序号',
    dataIndex: 'number',
    width: '60px',
    align: 'center',
    scopedSlots: { customRender: 'number' }
  },
  {
    title: '位置信息',
    dataIndex: 'address',
    width: '200px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '所属部门',
    dataIndex: 'departmentName',
    width: '180px',
    align: 'center'
  },
  {
    title: '生产负责人',
    dataIndex: 'productChargerName',
    width: '180px',
    align: 'center'
  },
  {
    title: '消防负责人',
    dataIndex: 'chargePerson',
    width: '180px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '状态',
    dataIndex: 'statusName',
    width: '120px',
    align: 'center',
    scopedSlots: { customRender: 'statusName' }
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '180px',
    align: 'center',
    // fixed: 'right',
    scopedSlots: { customRender: 'action' }
  }
]

export const firePlanColumns = [
  {
    title: '序号',
    dataIndex: 'number',
    width: '60px',
    align: 'center',
    scopedSlots: { customRender: 'number' }
  },
  {
    title: '计划名称',
    dataIndex: 'name',
    width: '200px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '计划类型',
    dataIndex: 'type',
    width: '180px',
    align: 'center',
    scopedSlots: { customRender: 'type' }
  },
  {
    title: '检查开始日期',
    dataIndex: 'startDate',
    width: '180px',
    align: 'center'
  },
  {
    title: '检查结束日期',
    dataIndex: 'endDate',
    width: '180px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '负责人',
    dataIndex: 'chargerName',
    width: '180px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '状态',
    dataIndex: 'status',
    width: '120px',
    align: 'center',
    scopedSlots: { customRender: 'statusName' }
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '180px',
    align: 'center',
    // fixed: 'right',
    scopedSlots: { customRender: 'action' }
  }
]

export const firePlanItemColumns = [
  {
    title: '序号',
    customRender: (text, record, index) => `${index + 1}`,
    dataIndex: 'number',
    width: '60px',
    align: 'center'
  },
  {
    title: '编号',
    dataIndex: 'number',
    width: '120px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '消防器材名称',
    dataIndex: 'name',
    width: '150px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '规格型号',
    dataIndex: 'models',
    width: '120px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '位置',
    dataIndex: 'address',
    width: '150px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '排查人',
    dataIndex: 'planCheckPersonList',
    // width: '320px',
    align: 'center',
    scopedSlots: { customRender: 'planCheckPersonList' }
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '180px',
    align: 'center',
    scopedSlots: { customRender: 'action' }
  }
]

export const firePlanExecuteColumns = [
  {
    title: '序号',
    dataIndex: 'number',
    width: '60px',
    align: 'center',
    scopedSlots: { customRender: 'number' }
  },
  {
    title: '排查计划名称',
    dataIndex: 'name',
    width: '300px',
    align: 'center'
  },
  {
    title: '排查类型',
    dataIndex: 'type',
    width: '140px',
    align: 'center',
    scopedSlots: { customRender: 'type' }
  },
  {
    title: '负责人',
    dataIndex: 'chargerName',
    width: '150px',
    align: 'center'
  },
  {
    title: '检查日期',
    dataIndex: 'startDate',
    width: '350px',
    align: 'center',
    customRender: (text, record, index) => `${record.startDate} ~ ${record.endDate}`
  },
  {
    title: '派发日期',
    dataIndex: 'dispatchDate',
    width: '200px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '状态',
    dataIndex: 'status',
    width: '120px',
    align: 'center',
    scopedSlots: { customRender: 'status' }
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '120px',
    align: 'center',
    scopedSlots: { customRender: 'action' }
  }
]

export const firePlanExecuteCheckColumns = [
  {
    title: '序号',
    customRender: (text, record, index) => `${index + 1}`,
    dataIndex: 'number',
    width: '60px',
    align: 'center'
  },
  {
    title: '编号',
    dataIndex: 'number',
    align: 'center',
    ellipsis: true
  },
  {
    title: '消防器材名称',
    dataIndex: 'name',
    align: 'center',
    ellipsis: true
  },
  {
    title: '规格型号',
    dataIndex: 'models',
    align: 'center',
    ellipsis: true
  },
  {
    title: '位置',
    dataIndex: 'address',
    align: 'center',
    ellipsis: true
  },
  {
    title: '排查人',
    dataIndex: 'planCheckPersonList',
    align: 'center',
    width: '200px',
    scopedSlots: { customRender: 'planCheckPersonList' }
  },
  {
    title: '状态',
    dataIndex: 'status',
    align: 'center',
    width: '80px',
    scopedSlots: { customRender: 'status' }
  },
  {
    title: '操作',
    dataIndex: 'action',
    align: 'center',
    scopedSlots: { customRender: 'action' }
  }
]

export const fireEquipmentAbnormalityColumns = [
  {
    title: '序号',
    dataIndex: 'number',
    width: '60px',
    align: 'center',
    scopedSlots: { customRender: 'number' }
  },
  {
    title: '编号',
    dataIndex: 'number',
    width: '120px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '消防器材名称',
    dataIndex: 'name',
    width: '150px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '规格型号',
    dataIndex: 'models',
    width: '150px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '位置',
    dataIndex: 'address',
    width: '160px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '处理部门',
    dataIndex: 'processorDeptName',
    width: '120px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '检查时间',
    dataIndex: 'processDate',
    width: '180px',
    align: 'center'
  },
  {
    title: '提交人',
    dataIndex: 'registrantName',
    width: '150px',
    align: 'center'
  },
  {
    title: '提交部门',
    dataIndex: 'registrantDeptName',
    width: '150px',
    align: 'center'
  },
  {
    title: '状态',
    dataIndex: 'status',
    width: '120px',
    align: 'center',
    scopedSlots: { customRender: 'status' }
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '80px',
    align: 'center',
    scopedSlots: { customRender: 'action' }
  }
]
