// spang-web/src/views/safety/risk/columns.js
// 设备风险点清单
export const equipmentColumns = [
  {
    title: '序号',
    scopedSlots: {
      customRender: 'serial'
    },
    dataIndex: 'index',
    width: '60px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '风险点编号',
    dataIndex: 'equipmentCode',
    width: '120px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '设备设施名称',
    dataIndex: 'equipmentName',
    width: '120px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '设备设施类型',
    dataIndex: 'equipmentTypeName',
    width: '120px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '区域/位置',
    dataIndex: 'areaLocation',
    width: '150px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '管辖部门/单位',
    dataIndex: 'deptName',
    width: '150px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '是否特种设备',
    dataIndex: 'isSpecial',
    customRender: (text, record) => (record.isSpecial ? '否' : '是'),
    width: '150px',
    align: 'center'
  },
  {
    title: '风险级别',
    dataIndex: 'riskLevelName',
    width: '100px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '风险类型',
    dataIndex: 'riskType',
    width: '100px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '状态',
    dataIndex: 'status',
    width: '80px',
    align: 'center',
    scopedSlots: { customRender: 'status' }
    // customRender: (text, record) => {
    //   const statusMap = { 0: '未提交', 1: '已提交', 2: '已审批' }
    //   return statusMap[record.status]
    // }
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '220px',
    align: 'center',
    // fixed: 'right',
    scopedSlots: { customRender: 'action' }
  }
]
// 设备风险点列表
export const equipmentCheckColumns = [
  {
    title: '序号',
    customRender: (text, record, index) => `${index + 1}`,
    dataIndex: 'index',
    width: '80px',
    align: 'center'
  },
  {
    title: '检查项目',
    dataIndex: 'checkName',
    align: 'center',
    scopedSlots: { customRender: 'checkName' }
  },
  {
    title: '检查标准',
    dataIndex: 'checkStandard',
    align: 'center',
    scopedSlots: { customRender: 'checkStandard' }
  },
  {
    title: '不符合标准情况',
    dataIndex: 'nonCompliance',
    align: 'center',
    scopedSlots: { customRender: 'nonCompliance' }
  },
  {
    title: '主要后果',
    dataIndex: 'mainConsequence',
    align: 'center',
    scopedSlots: { customRender: 'mainConsequence' }
  },
  {
    title: '操作项',
    dataIndex: 'action',
    width: '200px',
    align: 'center',
    fixed: 'right',
    scopedSlots: { customRender: 'action' }
  }
]
// 设备风险点风险评价LS
export const LSCoumns = [
  {
    title: '等级',
    dataIndex: 'seriousScores',
    align: 'center'
  },
  {
    title: '描述',
    dataIndex: 'seriousDetails',
    align: 'center'
  }
]

// 设备风险点风险评价R
export const RColumns = [
  {
    title: '风险度',
    dataIndex: 'seriousScores',
    align: 'center',
    customRender: (text, record) => {
      const data = [record.judgeMin, record.judgeMax].filter(item => (item != null && item != undefined && item != -1))
      return data.join('-')
    }
  },
  {
    title: '级别',
    dataIndex: 'judgeLevelName',
    align: 'center'
  },
  {
    title: '应采取的行动/控制措施',
    dataIndex: 'controlMeasures',
    align: 'center',
    ellipsis: true
  },
  {
    title: '实施期限',
    dataIndex: 'timeLimit',
    align: 'center'
  }
]

// L判定准则
export const LColumns = [
  {
    title: '分数',
    dataIndex: 'scores',
    align: 'center'
  },
  {
    title: '事件发生频率',
    dataIndex: 'happenFrequency',
    align: 'center',
    ellipsis: true
  },
  {
    title: '安全检查',
    dataIndex: 'securityCheck',
    align: 'center',
    ellipsis: true
  },
  {
    title: '操作规程',
    dataIndex: 'workRule',
    align: 'center',
    ellipsis: true
  },
  {
    title: '员工胜任程度（意识、技能、经验）',
    dataIndex: 'employeeCompetency',
    align: 'center',
    ellipsis: true
  },
  {
    title: '防范、控制措施',
    dataIndex: 'controlMeasures',
    align: 'center',
    ellipsis: true
  }
]

// S判定准则
export const SColumns = [
  {
    title: '分数',
    dataIndex: 'scores',
    align: 'center'
  },
  {
    title: '人员伤亡程度',
    dataIndex: 'casualties',
    align: 'center',
    ellipsis: true
  },
  {
    title: '直接财产损失',
    dataIndex: 'propertyLoss',
    align: 'center',
    ellipsis: true
  },
  {
    title: '停工时间',
    dataIndex: 'downtime',
    align: 'center',
    ellipsis: true
  },
  {
    title: '对声誉的影响',
    dataIndex: 'reputationImpact',
    align: 'center',
    ellipsis: true
  }
]

// 作业活动风险主列表
export const activityRiskColumns = [
  {
    title: '序号',
    scopedSlots: {
      customRender: 'serial'
    },
    dataIndex: 'index',
    width: '60px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '风险点编号',
    dataIndex: 'code',
    width: '150px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '活动类型',
    dataIndex: 'activityTypeName',
    width: '100px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '活动名称',
    dataIndex: 'activityName',
    width: '100px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '区域/位置',
    dataIndex: 'areaLocation',
    width: '150px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '管辖部门/单位',
    dataIndex: 'deptName',
    width: '150px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '是否特种作业',
    dataIndex: 'isSpecial',
    customRender: (text, record) => (record.isSpecial ? '否' : '是'),
    width: '150px',
    align: 'center'
  },
  {
    title: '风险级别',
    dataIndex: 'riskLevelName',
    width: '100px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '风险类型',
    dataIndex: 'riskType',
    width: '100px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '状态',
    dataIndex: 'status',
    width: '80px',
    align: 'center',
    scopedSlots: { customRender: 'status' }
    // customRender: (text, record) => {
    //   const statusMap = { 0: '未提交', 1: '已提交', 2: '已审批' }
    //   return statusMap[record.status]
    // }
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '220px',
    align: 'center',
    // fixed: 'right',
    scopedSlots: { customRender: 'action' }
  }
]

export const activityCheckColumns = [
  {
    title: '序号',
    customRender: (text, record, index) => `${index + 1}`,
    dataIndex: 'index',
    width: '80px',
    align: 'center'
  },
  {
    title: '作业步骤',
    dataIndex: 'workFlow',
    align: 'center',
    scopedSlots: { customRender: 'workFlow' }
  },
  {
    title: '危险源或潜在事件',
    dataIndex: 'dangerousSource',
    align: 'center',
    scopedSlots: { customRender: 'dangerousSource' }
  },
  {
    title: '可能发生的事故类型及后果',
    dataIndex: 'possibleConsequences',
    align: 'center',
    scopedSlots: { customRender: 'possibleConsequences' }
  },
  {
    title: '操作项',
    dataIndex: 'action',
    width: '200px',
    align: 'center',
    fixed: 'right',
    scopedSlots: { customRender: 'action' }
  }
]
