// spang-web/src/views/safety/potentialrisk/checkoutManageMgr/table.js
export const potentialPlanColumns = [{
    title: '计划名称',
    dataIndex: 'name',
    align: 'center',
    width: '150px',
  },
  {
    title: '排查类别',
    dataIndex: 'typeName',
    width: '150px',
    align: 'center',
  },
  {
    title: '排查人',
    dataIndex: 'checkPlanCheckPersonList',
    width: '150px',
    align: 'center',
    scopedSlots: {
      customRender: 'checkPlanCheckPersonList'
    },
  },
  {
    title: '排查开始日期',
    dataIndex: 'startTime',
    width: '150px',
    align: 'center',
    scopedSlots: {
      customRender: 'startTime'
    },
  },
  {
    title: '排查结束日期',
    dataIndex: 'endTime',
    width: '150px',
    align: 'center',
    scopedSlots: {
      customRender: 'endTime'
    },
  },
  {
    title: '派发日期',
    dataIndex: 'distributeTime',
    width: '150px',
    align: 'center',
    scopedSlots: {
      customRender: 'distributeTime'
    },
  },
  {
    title: '状态',
    dataIndex: 'status',
    align: 'center',
    width: '100px',
    scopedSlots: {
      customRender: 'status'
    },
  },
  {
    title: '操作',
    dataIndex: 'action',
    align: 'center',
    width: '150px',
    scopedSlots: {
      customRender: 'action'
    },
  },
]

export const columnsPotentialFactor = [{
    title: '风险点',
    dataIndex: 'perilPointName',
    width: '100px',
    align: 'center',
    ellipsis: true,
  },
  {
    title: '风险类型',
    dataIndex: 'type',
    width: '150px',
    align: 'center',
    ellipsis: true,
  },
  {
    title: '管控层级',
    dataIndex: 'controlLevel',
    width: '100px',
    align: 'center',
    ellipsis: true,
  },
  {
    title: '位置/区域',
    dataIndex: 'address',
    width: '100px',
    align: 'center',
    ellipsis: true,
  },
  {
    title: '排查人',
    dataIndex: 'checkPerson',
    width: '100px',
    align: 'center',
    ellipsis: true,
    scopedSlots: {
      customRender: 'checkPerson'
    },
  },
  {
    title: '检查项',
    dataIndex: 'checkItems',
    scopedSlots: {
      customRender: 'checkItems'
    },
    align: 'center',
  },
  {
    title: '状态',
    dataIndex: 'status',
    scopedSlots: {
      customRender: 'status'
    },
    width: '100px',
    align: 'center',
    ellipsis: true,
  },
  {
    title: '操作',
    dataIndex: 'action',
    align: 'center',
    scopedSlots: {
      customRender: 'action'
    },
  },
]

// 设备设施类表头
export const columnsFacility = [{
    title: '序号',
    customRender: (text, record, index) => `${index + 1}`,
    dataIndex: 'number',
    align: 'center',
    width: '80px',
  }, {
    title: '检查项目',
    dataIndex: 'checkName',
    align: 'center',

  },
  {
    title: '检查标准',
    dataIndex: 'checkStandard',
    align: 'center',

  },
  {
    title: '不符合标准情况',
    dataIndex: 'nonCompliance',
    align: 'center',

  },
  {
    title: '主要后果',
    dataIndex: 'mainConsequence',
    align: 'center',

  },
]

// 作业活动类表头
export const columnsTask = [{
    title: '序号',
    customRender: (text, record, index) => `${index + 1}`,
    dataIndex: 'number',
    align: 'center',
    width: '80px',
  }, {
    title: '作业步骤',
    dataIndex: 'workFlow',
    align: 'center',
  },
  {
    title: '危险源或潜在事件',
    dataIndex: 'dangerousSource',
    align: 'center',
  },
  {
    title: '可能发生的事故类型及后果',
    dataIndex: 'possibleConsequences',
    align: 'center',
  },
]
