// spang-web/src/views/safety/task/blind-plate/risks.js
export const hotWorkRiskFillinList = {
  riskIdentify: [
    {
      label: '爆炸性气体',
      checked: false
    },
    {
      label: '易燃性物质',
      checked: false
    },
    {
      label: '腐蚀性液体',
      checked: false
    },
    {
      label: '有毒有害化学品',
      checked: false
    },
    {
      label: '高压气体/液体',
      checked: false
    },
    {
      label: '冻伤',
      checked: false
    },
    {
      label: '爆炸性粉尘',
      checked: false
    },
    {
      label: '窒息',
      checked: false
    },
    {
      label: '危险物料混窜',
      checked: false
    },
    {
      label: '物料自燃',
      checked: false
    },
    {
      label: '不利天气',
      checked: false
    },
    {
      label: '化学反应',
      checked: false
    },
    {
      label: '隐蔽工程泄漏',
      checked: false
    },
    {
      label: '灼伤/烫伤',
      checked: false
    },
    {
      label: '坠落',
      checked: false
    },
    {
      label: '产生火花',
      checked: false
    },
    {
      label: '淹溺/坍塌',
      checked: false
    },
    {
      label: '触电/静电',
      checked: false
    },
    {
      label: '湿滑跌倒',
      checked: false
    },
    {
      label: '物体打击',
      checked: false
    },
    {
      label: '中毒',
      checked: false
    },
    {
      label: '其他',
      checked: false,
      content: ' '
    }
  ],
  specification: [
    {
      label: '在易燃易爆场所作业，作业地点30米内不得有动火作业',
      checked: false
    },
    {
      label: '严禁在同一管道上同时进行两处及两处以上盲板抽堵作业',
      checked: false
    },
    {
      label: '出现危险品泄漏时，立即停止作业，撤离人员',
      checked: false
    },
    {
      label: '人员培训已完成',
      checked: false
    }
  ],
  measure: [
    {
      label: '切断工艺流程',
      checked: false
    },
    {
      label: '隔离、吹扫、置换',
      checked: false
    },
    {
      label: '工作警示牌',
      checked: false
    },
    {
      label: '设路障',
      checked: false
    },
    {
      label: '安全冲洗设施',
      checked: false
    },
    {
      label: '电源切断',
      checked: false
    },
    {
      label: '通讯工具',
      checked: false
    },
    {
      label: '气体检测仪',
      checked: false
    },
    {
      label: '通风',
      checked: false
    },
    {
      label: '夜间照明和警示灯具',
      checked: false
    },
    {
      label: '气体检测',
      checked: false
    },
    {
      label: '防爆机具',
      checked: false
    },
    {
      label: '急救设施',
      checked: false
    },
    {
      label: '消防设施',
      checked: false
    },
    {
      label: '工作区域围栏、警戒线',
      checked: false
    },
    {
      label: '其他措施',
      checked: false,
      content: ' '
    }
  ],
  equipment: [
    {
      label: '全封闭眼罩',
      checked: false
    },
    {
      label: '安全帽',
      checked: false
    },
    {
      label: '防静电服装',
      checked: false
    },
    {
      label: '防酸碱服',
      checked: false
    },
    {
      label: '防毒面罩',
      checked: false
    },
    {
      label: '安全带',
      checked: false
    },
    {
      label: '正压式呼吸器',
      checked: false
    },
    {
      label: '防化服',
      checked: false
    },
    {
      label: '手套',
      checked: false
    },
    {
      label: '其他个人防护装备',
      checked: false,
      content: ' '
    }
  ]
}
