// spang-web/src/views/safety/orientation/threeSafeEdu/table.js
export const educationColumns = [
  // {
  //   title: '序号',
  //   customRender: (text, record, index) => `${index + 1}`,
  //   dataIndex: 'number',
  //   width: '100px',
  // },
  {
    title: '员工',
    dataIndex: 'userName',
    align: 'center',
    width: '150px',
    ellipsis: true
  },
  {
    title: '培训单位',
    dataIndex: 'trainDeptName',
    align: 'center',
    width: '200px',
    ellipsis: true
  },
  {
    title: '教育类型',
    dataIndex: 'eduTypeName',
    align: 'center',
    width: '150px',
    ellipsis: true
  },
  {
    title: '培训课时',
    dataIndex: 'trainPeriod',
    width: '100px',
    align: 'center'
  },

  {
    title: '培训日期',
    dataIndex: 'teachStartDate',
    width: '200px',
    ellipsis: true,
    align: 'center'
  },
  {
    title: '考试成绩',
    dataIndex: 'examScore',
    width: '100px',
    align: 'center'
  },
  {
    title: '考试结果',
    dataIndex: 'examResult',
    align: 'center',
    width: '100px',
    scopedSlots: {
      customRender: 'examResult'
    }
  },

  {
    title: '状态',
    dataIndex: 'status',
    align: 'center',
    width: '100px',
    scopedSlots: {
      customRender: 'status'
    }
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '250px',
    align: 'center',
    scopedSlots: {
      customRender: 'action'
    }
  }
]
