// spang-web/src/router/static-router.js
// eslint-disable-next-line
import { UserLayout, BasicLayout, RouteView, BlankLayout } from '@/layouts'
/**
 * 基础路由
 * @type { *[] }
 */
export const constantRouterMap = [
  {
    path: '/user',
    component: UserLayout,
    redirect: '/user/login',
    hidden: true,
    children: [
      {
        path: 'login',
        name: 'login',
        component: () => import(/* webpackChunkName: "user" */ '@/views/user/Login')
      },
      {
        path: 'register',
        name: 'register',
        component: () => import(/* webpackChunkName: "user" */ '@/views/user/Register')
      },
      {
        path: 'register-result',
        name: 'registerResult',
        component: () => import(/* webpackChunkName: "user" */ '@/views/user/RegisterResult')
      },
      {
        path: 'recover',
        name: 'recover',
        component: undefined
      }
    ]
  },
  {
    path: '/',
    name: 'index',
    redirect: '/system',
    component: undefined
  },
  {
    path: '/system',
    name: 'system',
    component: BasicLayout,
    redirect: '/system/workplace',
    hidden: true,
    meta: { title: '智能管理平台' },
    children: [
      {
        path: '/system/workplace',
        name: 'workplace',
        component: () => import('@/views/system/index1'),
        meta: { title: '工作台', icon: 'appstore', hiddenHeaderContent: true }
      },
      {
        path: '/system/index1',
        name: 'index1',
        component: () => import('@/views/system/index'),
        meta: { title: '菜单一', icon: 'menu' }
      },
      {
        path: '/system/index2',
        name: 'index2',
        component: undefined,
        meta: { title: '菜单二', icon: 'menu' }
      },
      {
        path: '/system/index3',
        name: 'index3',
        component: undefined,
        meta: { title: '菜单三', icon: 'menu' }
      },
      {
        path: '/system/index4',
        name: 'index4',
        component: undefined,
        meta: { title: '菜单四', icon: 'menu' }
      }
    ]
  },
  {
    path: '/emergency',
    name: 'emergency',
    component: BasicLayout,
    meta: { title: '应急管理' },
    redirect: '/emergency/resources',
    children: [
      {
        path: '/emergency/linkage',
        name: 'linkage',
        component: () => import('@/views/emergency/linkage/index'),
        meta: { title: '应急联动', icon: 'deployment-unit' }
      },
      {
        path: '/emergency/agency',
        name: 'agency',
        component: RouteView,
        redirect: '/emergency/agency/firefighting',
        meta: { title: '应急机构', icon: 'bank' },
        children: [
          {
            path: '/emergency/agency/firefighting',
            name: 'firefighting',
            component: () => import('@/views/emergency/agency/firefighting'),
            meta: { title: '消防应急机构', keepAlive: true }
          },
          {
            path: '/emergency/agency/medical',
            name: 'medical',
            component: () => import('@/views/emergency/agency/medical'),
            meta: { title: '医疗保障机构', keepAlive: true }
          },
          {
            path: '/emergency/agency/rescue',
            name: 'rescue',
            component: () => import('@/views/emergency/agency/rescue'),
            meta: { title: '应急救援保障机构', keepAlive: true }
          },
          {
            path: '/emergency/agency/communication',
            name: 'communication',
            component: () => import('@/views/emergency/agency/communication'),
            meta: { title: '通讯保障机构', keepAlive: true }
          },
          {
            path: '/emergency/agency/technology',
            name: 'technology',
            component: () => import('@/views/emergency/agency/technology'),
            meta: { title: '技术支持机构', keepAlive: true }
          }
        ]
      },
      {
        path: '/emergency/resources',
        name: 'resources',
        component: RouteView,
        redirect: '/emergency/resources/supplies',
        meta: { title: '应急资源', icon: 'hourglass' },
        // component: () => import('@/views/emergency/resources/resources'),
        children: [
          {
            path: '/emergency/resources/supplies',
            name: 'supplies',
            component: () => import('@/views/emergency/resources/supplies'),
            meta: { title: '应急物资装备', keepAlive: true }
          },
          {
            path: '/emergency/resources/experts',
            name: 'experts',
            component: () => import('@/views/emergency/resources/experts'),
            meta: { title: '应急专家', keepAlive: true }
          },
          {
            path: '/emergency/resources/staffs',
            name: '应急人员',
            component: () => import('@/views/emergency/resources/staffs'),
            meta: { title: '应急人员', keepAlive: true }
          },
          {
            path: '/emergency/resources/vehicle',
            name: '应急车辆',
            component: () => import('@/views/emergency/resources/vehicle'),
            meta: { title: '应急车辆', keepAlive: true }
          }
        ]
      },
      {
        path: '/emergency/plans',
        name: 'plans',
        component: RouteView,
        meta: { title: '应急预案', icon: 'file-protect' }
      },
      {
        path: '/emergency/rescues',
        name: 'rescues',
        component: RouteView,
        meta: { title: '应急救援', icon: 'sound' }
      },
      {
        path: '/emergency/duty',
        name: 'duty',
        component: RouteView,
        meta: { title: '应急值守', icon: 'user' }
      }
    ]
  },
  {
    path: '/safety',
    name: 'safety',
    component: BasicLayout,
    meta: { title: '智慧安监' },
    redirect: '/safety/important',
    children: [
      {
        path: '/safety/overview',
        name: 'overview',
        component: () => import('@/views/safety/overview/overview'),
        meta: { title: '安全态势总览' }
      },
      {
        path: '/safety/important',
        name: 'important',
        component: () => import('@/views/safety/important/index'),
        meta: { title: '两重点一重大' }
      },
      {
        path: '/safety/forewarning',
        name: 'forewarning',
        component: () => import('@/views/safety/forewarning/index'),
        meta: { title: '安全预警管理' }
      },
      {
        path: '/safety/organizations',
        name: 'organizations',
        component: () => import('@/views/safety/organizations/index'),
        meta: { title: '安全管理机构和人员' }
      },
      {
        path: '/safety/law',
        name: 'law',
        component: () => import('@/views/safety/law/law'),
        meta: { title: '法律法规' }
      },
      {
        path: '/safety/rules',
        name: 'rules',
        component: undefined,
        meta: { title: '安全管理制度' }
      },
      {
        path: '/safety/orientation',
        name: 'orientation',
        component: undefined,
        meta: { title: '安全教育培训' }
      },
      {
        path: '/safety/schedule',
        name: 'schedule',
        component: undefined,
        meta: { title: '安全工作计划' }
      },
      {
        path: '/safety/commitment',
        name: 'commitment',
        component: undefined,
        meta: { title: '安全承诺' }
      },
      {
        path: '/safety/risk',
        name: 'risk',
        component: undefined,
        meta: { title: '风险管理' }
      },
      {
        path: '/safety/potentialrisk',
        name: 'potentialrisk',
        component: undefined,
        meta: { title: '隐患管理' }
      },
      // {
      //   path: '/safety/work',
      //   name: 'work',
      //   component: undefined,
      //   meta: { title: '作业管理' }
      // },
      {
        path: '/safety/contractor',
        name: 'contractor',
        component: undefined,
        meta: { title: '承包商管理' }
      },
      {
        path: '/safety/alteration',
        name: 'orientation',
        component: undefined,
        meta: { title: '变更管理' }
      },
      {
        path: '/safety/accident',
        name: 'accident',
        component: () => import('@/views/safety/accident/AccidentManagement'),
        meta: { title: '事故管理' }
      },
      {
        path: '/safety/health',
        name: 'health',
        component: () => import('@/views/safety/culture/index'),
        meta: { title: '职业卫生管理' }
      },
      {
        path: '/safety/cost',
        name: 'cost',
        component: () => import('@/views/safety/culture/index'),
        meta: { title: '安全费用管理' }
      },
      {
        path: '/safety/culture',
        name: 'culture',
        component: () => import('@/views/safety/culture/index'),
        meta: { title: '安全文化管理' }
      }
    ]
  },
  {
    path: '/environment',
    name: 'environment',
    component: BasicLayout,
    meta: { title: '智慧环保' },
    redirect: '/environment/warning',
    children: [
      {
        path: '/environment/overview',
        name: 'overview1',
        // component: () => import('@/views/environment/overview/overview'),
        meta: { title: '环保态势总览' }
      },
      {
        path: '/environment/atmosphere',
        name: 'atmosphere',
        // component: () => import('@/views/environment/overview/overview'),
        meta: { title: '大气环境综合管理' }
      },
      {
        path: '/environment/water',
        name: 'water',
        // component: () => import('@/views/environment/water/index'),
        meta: { title: '水环境综合管理' }
      },
      {
        path: '/environment/noise',
        name: 'noise',
        // component: () => import('@/views/environment/noise/index'),
        meta: { title: '噪声管理' }
      },
      {
        path: '/environment/waste',
        name: 'waste',
        component: () => import('@/views/environment/waste/index'),
        meta: { title: '固危废管理' }
      },
      {
        path: '/environment/status',
        name: 'status',
        component: undefined,
        meta: { title: '企业工况管理' }
      },
      {
        path: '/environment/warning',
        name: 'warning',
        component: () => import('@/views/environment/warning/index'),
        meta: { title: '环保预报警管理' }
      }
    ]
  },
  {
    path: '/404',
    component: () => import(/* webpackChunkName: "fail" */ '@/views/exception/404')
  },
  {
    path: '*', redirect: '/404', hidden: true
  }
]
