// spang-web/src/views/safety/task/confined-space/risks.js
export const hotWorkRiskFillinList = [
    {
      titleId: 1,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '作业人员不清楚现场危险',
      measureLabel: '作业前进行安全教育'
    },
    {
      titleId: 2,
      isTitle: 1,
      titleValue: '',
      isMeasures: 0,
      label: '系统未隔绝',
      measureLabel: '',
      measuresValue: '',
      showCheckbox: true,
      measuresText: '',
      options: [
        {
          label: '所有与受限空间联系的管线、阀门加盲板隔离，列出盲板清单，落实责任人',
          value: 0
        }, {
          label: '拆除管道或法兰',
          value: 1
        }
      ]
    },
    {
      titleId: 3,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      measuresValue: '',
      measuresText: '',
      label: '通风不好',
      measureLabel: '设备打开通风孔自然通风，温度适宜人员作业；必要时采取强制通风或佩戴空气呼吸器，不能用通氧气或富氧空气的方法补充氧。'
    },
    {
      titleId: 4,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '系统或设备内存危化品',
      measureLabel: '进行置换、吹扫、蒸煮、冲洗至分析（提前30分钟）合格，涂刷具有挥发性溶剂的涂料时应连续分析。'
    },
    {
      titleId: 5,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '存在搅拌、转动等设备',
      measureLabel: '相关设备进行处理，带搅拌机设备切断电源，电源开关处加锁或挂“禁止合闸”标志牌，设专人监护。'
    },
    {
      titleId: 6,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '易燃易爆环境',
      measureLabel: '使用防爆低压灯具（干燥容器内为36V，潮湿或狭小容器内为12V）和防爆工具，禁止使用可能产生火花的工具。'
    },
    {
      titleId: 7,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '受限空间进出口通道不畅',
      measureLabel: '检查受限空间进出口通道，不得有阻碍人员进出的障碍物'
    },
    {
      titleId: 8,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '盛装过可燃有毒液体、气体的受限空间',
      measureLabel: '分析盛装过可燃有毒液体、气体的受限空间内的可燃、有毒有害气体含量。'
    },
    {
      titleId: 9,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      measuresValue: [],
      measuresText: '',
      label: '不佩戴劳动防护用品及未准备应急用品',
      measureLabel: '按规定佩戴安全带（绳）、防毒用品等，备有空气呼吸器、消防器材或清水等应急用品'
    },
    {
      titleId: 10,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '监护不到位',
      measureLabel: '指派专人监护，并坚守岗位；险情重大的作业，应增设监护人员'
    },
    {
      titleId: 11,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '使用的设备、工具不安全',
      measureLabel: '检查，确保安全可靠。'
    },
    {
      titleId: 12,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '内外人员联络不畅',
      measureLabel: '正常作业时，内外可通过绳索互通信号或配备可靠的通讯工具。'
    },
    {
      titleId: 13,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '无事故情况下的应急措施',
      measureLabel: '工作者感到不适时，要连续不断地扯动绳索或使用通讯工具报告，并在监护人协助下离开。发生事故时监护人要立即报告，救护人员必须做好自身防护方可入内实施抢救。'
    },
    {
      titleId: 14,
      isTitle: 1,
      titleValue: '',
      isMeasures: 0,
      measuresValue: [],
      measuresText: '',
      label: '作业监护措施',
      measureLabel: '',
      options: [
        {
          label: '灭火器$具'
        }, {
          label: '生化服$身'
        }, {
          label: '空气呼吸器$个'
        }, {
          label: '长管$个'
        }, {
          label: '防毒面罩$个'
        }, {
          label: '其他$个'
        }
      ]
    },
    {
      titleId: 15,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '作业人员违反安全规程',
      measureLabel: '监护人应立即令其停止工作。'
    },
    {
      titleId: 16,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '进入下水井、管道、池时',
      measureLabel: '下水井、管道、池中污泥含有硫化氢或其他毒物时，应佩戴安全带（绳），防毒面具等。'
    },
    {
      titleId: 17,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '作业后现场留有杂物',
      measureLabel: '清理杂物。'
    },
    {
      titleId: 18,
      isTitle: 1,
      titleValue: '',
      isMeasures: 0,
      measuresText: '',
      label: '其他措施',
      measureLabel: 'false'
    }
  ]
