// spang-web/src/views/safety/potentialrisk/components/table.js
// L判定准则
export const LColumns = [{
    title: '分数',
    dataIndex: 'scores',
    align: 'center'
  },
  {
    title: '事件发生频率',
    dataIndex: 'happenFrequency',
    align: 'center',
    ellipsis: true
  },
  {
    title: '安全检查',
    dataIndex: 'securityCheck',
    align: 'center',
    ellipsis: true
  },
  {
    title: '操作规程',
    dataIndex: 'workRule',
    align: 'center',
    ellipsis: true
  },
  {
    title: '员工胜任程度（意识、技能、经验）',
    dataIndex: 'employeeCompetency',
    align: 'center',
    ellipsis: true
  },
  {
    title: '防范、控制措施',
    dataIndex: 'controlMeasures',
    align: 'center',
    ellipsis: true
  }
]

// S判定准则
export const SColumns = [{
    title: '分数',
    dataIndex: 'scores',
    align: 'center'
  },
  {
    title: '人员伤亡程度',
    dataIndex: 'casualties',
    align: 'center',
    ellipsis: true
  },
  {
    title: '直接财产损失',
    dataIndex: 'propertyLoss',
    align: 'center',
    ellipsis: true
  },
  {
    title: '停工时间',
    dataIndex: 'downtime',
    align: 'center',
    ellipsis: true
  },
  {
    title: '对声誉的影响',
    dataIndex: 'reputationImpact',
    align: 'center',
    ellipsis: true
  }
]
// 设备风险点风险评价R
export const RColumns = [{
    title: '风险度',
    dataIndex: 'seriousScores',
    align: 'center',
    width: '15%',
    customRender: (text, record) => {
      const data = [record.judgeMin, record.judgeMax].filter(item => (item != null && item != undefined && item != -1))
      return data.join('-')
    }
  },
  {
    title: '级别',
    dataIndex: 'judgeLevelName',
    align: 'center',
    width: '15%',
  },
  {
    title: '应采取的行动/控制措施',
    dataIndex: 'controlMeasures',
    align: 'center',
    ellipsis: true
  },
  {
    title: '实施期限',
    dataIndex: 'timeLimit',
    align: 'center',
    width: '25%',
  }
]
