// spang-web/src/views/environment/table.js
export const warningColumns = [
    {
      title: '事件描述',
      dataIndex: 'description',
      width: '400px'
    },
    {
      title: '报警级别',
      dataIndex: 'level',
      align: 'center'
    },
    {
      title: '报警类型',
      dataIndex: 'type',
      align: 'center'
    },
    {
      title: '报警子类型',
      dataIndex: 'stype',
      align: 'center'
    },
    {
      title: '报警点位名称',
      dataIndex: 'point',
      align: 'center'
    },
    {
        title: '监测值',
        dataIndex: 'monitor',
        align: 'center'
      },
    {
      title: '阈值',
      dataIndex: 'threshold',
      align: 'center'
    },
    {
        title: '报警时间',
        dataIndex: 'time',
        align: 'center'
      },
      {
        title: '状态',
        dataIndex: 'status',
        align: 'center'
      },
    {
      title: '是否有效',
      dataIndex: 'effective',
      align: 'center',
      scopedSlots: { customRender: 'effective' }
    },
    {
      title: '操作',
      dataIndex: 'action',
      width: '150px',
      align: 'center',
      // fixed: true,
      scopedSlots: { customRender: 'action' }
    }
  ]
  export const personnelColumns = [
    {
      title: '公司名称',
      dataIndex: 'company',
      align: 'center'
    },
    {
      title: '姓名',
      dataIndex: 'name',
      align: 'center'
    },
    {
      title: '性别',
      dataIndex: 'sex',
      align: 'center'
    },
    {
      title: '岗位',
      dataIndex: 'type',
      align: 'center'
    },
    {
      title: '联系方式',
      dataIndex: 'count',
      align: 'center'
    },
    {
      title: '详情',
      dataIndex: 'action',
      width: '150px',
      align: 'center',
      // fixed: true,
      scopedSlots: { customRender: 'action' }
    }
  ]
  export const cultureColumns = [
    {
      title: '序号',
      dataIndex: 'number',
      align: 'center'
    },
    {
      title: '公司名称',
      dataIndex: 'company',
      align: 'center'
    },
    {
      title: '资料名称',
      dataIndex: 'filename',
      align: 'center'
    },
    {
      title: '附件',
      dataIndex: 'appendix',
      align: 'center',
      scopedSlots: { customRender: 'appendix' }
    },
    {
      title: '操作',
      dataIndex: 'action',
      width: '150px',
      align: 'center',
      scopedSlots: { customRender: 'action' }
    }
  ]
