// spang-web/src/views/safety/orientation/personeducation/table.js
export const personEduColumns = [{
    title: '员工',
    dataIndex: 'name',
    // width: '200px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '部门',
    dataIndex: 'deptName',
    // width: '250px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '岗位',
    dataIndex: 'postName',
    // width: '250px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '查看档案',
    dataIndex: 'action',
    width: '150px',
    align: 'center',
    scopedSlots: {
      customRender: 'action'
    }
  }
]
// 三级安全教育培训信息
export const safetyEducationColumns = [{
    title: '序号',
    customRender: (text, record, index) => `${index + 1}`,
    dataIndex: 'number',
    width: '100px',
  },

  {
    title: '培训类型',
    dataIndex: 'eduTypeName',
    width: '150px',
    align: 'center'
  },
  {
    title: '培训时间',
    dataIndex: 'teachStartDate',
    width: '150px',
    align: 'center'
  },
  {
    title: '培训学时',
    dataIndex: 'trainPeriod',
    width: '150px',
    align: 'center',

  },
  {
    title: '考核结果',
    dataIndex: 'examResult',
    width: '100px',
    align: 'center',
    scopedSlots: {
      customRender: 'examResult'
    }

  }
]
// 证书信息
export const workCertListColumns = [{
    title: '序号',
    customRender: (text, record, index) => `${index + 1}`,
    dataIndex: 'number',
    width: '100px',
  },
  {
    title: '证书名称',
    dataIndex: 'certName',
    width: '150px',
    align: 'center'
  },
  {
    title: '证书编号',
    dataIndex: 'certCode',
    width: '150px',
    align: 'center'
  },
  {
    title: '发证机构',
    dataIndex: 'issuingOffice',
    width: '150px',
    align: 'center'
  },

  {
    title: '有效期至',
    dataIndex: 'certEndDate',
    width: '150px',
    align: 'center'
  },
  {
    title: '复审日期',
    dataIndex: 'recheckDate',
    width: '150px',
    align: 'center'
  },
  {
    title: '照片',
    dataIndex: 'action',
    width: '150px',
    align: 'center',
    scopedSlots: {
      customRender: 'action'
    }
  }
]
// 培训记录
export const trainingColumns = [{
    title: '序号',
    customRender: (text, record, index) => `${index + 1}`,
    dataIndex: 'number',
    width: '100px',
  },
  {
    title: '培训名称',
    dataIndex: 'trainName',
    width: '150px',
    align: 'center'
  },
  {
    title: '培训时间',
    dataIndex: 'trainTime',
    width: '300px',
    align: 'center',
    scopedSlots: {
      customRender: 'trainTime'
    }
  },
  {
    title: '岗位培训形式',
    dataIndex: 'trainWay',
    width: '150px',
    align: 'center',
    scopedSlots: {
      customRender: 'trainWay'
    }
  },
  {
    title: '学时',
    dataIndex: 'trainPeriod',
    width: '150px',
    align: 'center',
  }
]
// 考试记录
export const examUserListColumns = [{
    title: '序号',
    customRender: (text, record, index) => `${index + 1}`,
    dataIndex: 'number',
    width: '100px',
  },
  {
    title: '考试名称',
    dataIndex: 'examName',
    width: '150px',
    align: 'center'
  },
  {
    title: '考试时间',
    dataIndex: 'examTime',
    width: '150px',
    align: 'center'
  },
  {
    title: '考试形式',
    dataIndex: 'examShape',
    width: '150px',
    align: 'center',
    scopedSlots: {
      customRender: 'examShape'
    }
  },
  {
    title: '考试成绩',
    dataIndex: 'score',
    width: '150px',
    align: 'center'
  },
  {
    title: '考试结果',
    dataIndex: 'examResult',
    width: '100px',
    align: 'center',
    scopedSlots: {
      customRender: 'examResult'
    }

  },
]
// 三违记录
export const violationsListColumns = [{
    title: '序号',
    customRender: (text, record, index) => `${index + 1}`,
    dataIndex: 'number',
    width: '100px',
  },
  {
    title: '三违行为类型',
    dataIndex: 'behaviorType',
    width: '150px',
    align: 'center'
  },
  {
    title: '三违描述',
    dataIndex: 'violationsDesc',
    width: '150px',
    align: 'center'
  },
  {
    title: '违规发生时间',
    dataIndex: 'violationsData',
    width: '150px',
    align: 'center'
  },
  {
    title: '处理方式',
    dataIndex: 'processMode',
    width: '150px',
    align: 'center',

  }
]
