// spang-web/src/views/safety/accident/columns.js
export const accidentExpressColumns = [
  {
    title: '序号',
    dataIndex: 'number',
    width: '60px',
    align: 'center',
    scopedSlots: { customRender: 'number' }
  },
  {
    title: '事故名称',
    dataIndex: 'accidentName',
    width: '200px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '发生时间',
    dataIndex: 'occurrenceTime',
    width: '150px',
    align: 'center'
  },
  {
    title: '发生地点',
    dataIndex: 'occurrenceAddress',
    width: '150px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '事故简述',
    dataIndex: 'accidentReport',
    width: '200px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '上报人',
    dataIndex: 'reportUserName',
    width: '150px',
    align: 'center'
  },
  {
    title: '上报时间',
    dataIndex: 'reportTime',
    width: '150px',
    align: 'center'
  },
  {
    title: '状态',
    dataIndex: 'status',
    width: '120px',
    align: 'center',
    scopedSlots: { customRender: 'status' }
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '230px',
    align: 'center',
    // fixed: 'right',
    scopedSlots: { customRender: 'action' }
  }
]

export const accidentNearmissColumns = [
  {
    title: '序号',
    dataIndex: 'number',
    width: '60px',
    align: 'center',
    scopedSlots: { customRender: 'number' }
  },
  {
    title: '事件名称',
    dataIndex: 'accidentName',
    width: '200px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '主要原因',
    dataIndex: 'accidentType',
    width: '150px',
    align: 'center',
    scopedSlots: { customRender: 'accidentType' }
  },
  {
    title: '发生时间',
    dataIndex: 'occurrenceTime',
    width: '150px',
    align: 'center'
  },
  {
    title: '发生地点',
    dataIndex: 'occurrenceAddress',
    width: '200px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '责任部门',
    dataIndex: 'dutyDeptName',
    width: '150px',
    align: 'center'
  },
  {
    title: '上报人',
    dataIndex: 'reportUserName',
    width: '150px',
    align: 'center'
  },
  {
    title: '状态',
    dataIndex: 'status',
    width: '120px',
    align: 'center',
    scopedSlots: { customRender: 'status' }
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '230px',
    align: 'center',
    // fixed: 'right',
    scopedSlots: { customRender: 'action' }
  }
]

export const accidentReportColumns = [
  {
    title: '序号',
    dataIndex: 'number',
    width: '60px',
    align: 'center',
    scopedSlots: { customRender: 'number' }
  },
  {
    title: '事故名称',
    dataIndex: 'accidentName',
    width: '200px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '事故类型',
    dataIndex: 'accidentTypeName',
    width: '100px',
    align: 'center'
  },
  {
    title: '事故分级',
    dataIndex: 'accidentLevelName',
    width: '150px',
    align: 'center'
  },
  {
    title: '责任人',
    dataIndex: 'dutyUserName',
    width: '150px',
    align: 'center'
  },
  {
    title: '责任部门',
    dataIndex: 'dutyDeptName',
    width: '180px',
    align: 'center'
  },
  {
    title: '发生日期',
    dataIndex: 'occurrenceTime',
    width: '200px',
    align: 'center'
  },
  {
    title: '上报人',
    dataIndex: 'reportUserName',
    width: '150px',
    align: 'center'
  },
  {
    title: '状态',
    dataIndex: 'status',
    width: '120px',
    align: 'center',
    scopedSlots: { customRender: 'status' }
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '200px',
    align: 'center',
    // fixed: 'right',
    scopedSlots: { customRender: 'action' }
  }
]

export const accidentProcessColumns = [
  {
    title: '序号',
    dataIndex: 'number',
    width: '60px',
    align: 'center',
    scopedSlots: { customRender: 'number' }
  },
  {
    title: '事故名称',
    dataIndex: 'accidentName',
    width: '180px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '事故类型',
    dataIndex: 'accidentTypeName',
    width: '100px',
    align: 'center'
  },
  {
    title: '事故分级',
    dataIndex: 'accidentLevelName',
    width: '150px',
    align: 'center'
  },
  {
    title: '责任人',
    dataIndex: 'dutyUserName',
    width: '120px',
    align: 'center'
  },
  {
    title: '责任部门',
    dataIndex: 'dutyDeptName',
    width: '150px',
    align: 'center'
  },
  {
    title: '发生日期',
    dataIndex: 'occurrenceTime',
    width: '200px',
    align: 'center'
  },
  {
    title: '上报人',
    dataIndex: 'reportUserName',
    width: '150px',
    align: 'center'
  },
  {
    title: '查看事故',
    dataIndex: 'viewAccident',
    width: '100px',
    align: 'center',
    scopedSlots: { customRender: 'viewAccident' }
  },
  {
    title: '状态',
    dataIndex: 'isProcess',
    width: '120px',
    align: 'center',
    scopedSlots: { customRender: 'isProcess' }
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '120px',
    align: 'center',
    // fixed: 'right',
    scopedSlots: { customRender: 'action' }
  }
]

export const accidentArchiveColumns = [
  {
    title: '序号',
    dataIndex: 'number',
    width: '60px',
    align: 'center',
    scopedSlots: { customRender: 'number' }
  },
  {
    title: '事故名称',
    dataIndex: 'accidentName',
    width: '200px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '事故类型',
    dataIndex: 'accidentTypeName',
    width: '100px',
    align: 'center'
  },
  {
    title: '发生地点',
    dataIndex: 'occurrenceAddress',
    width: '150px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '损伤情况',
    dataIndex: 'peopleDamage',
    width: '150px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '事故内容',
    dataIndex: 'accidentReport',
    width: '150px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '发生时间',
    dataIndex: 'occurrenceTime',
    width: '180px',
    align: 'center'
  },
  // {
  //   title: '事故报告',
  //   dataIndex: '',
  //   width: '100px',
  //   align: 'center',
  //   scopedSlots: { customRender: 'files' }
  // },
  {
    title: '状态',
    dataIndex: 'isProcess',
    width: '120px',
    align: 'center',
    scopedSlots: { customRender: 'isProcess' }
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '120px',
    align: 'center',
    // fixed: 'right',
    scopedSlots: { customRender: 'action' }
  }
]
