// spang-web/src/views/safety/potentialrisk/planManageMgr/table.js
export const potentialPlanColumns = [{
    title: '计划名称',
    dataIndex: 'name',
    align: 'center',
    width: '250px',
    ellipsis: true
  },
  {
    title: '排查类别',
    dataIndex: 'typeName',
    width: '150px',
    align: 'center',
    ellipsis: true,
  },
  {
    title: '组织单位',
    dataIndex: 'organizeUnitName',
    width: '100px',
    align: 'center',
    ellipsis: true,
  },
  {
    title: '被检查单位',
    dataIndex: 'checkUnitName',
    width: '150px',
    align: 'center',
    ellipsis: true
  },

  {
    title: '排查开始日期',
    dataIndex: 'startTime',
    width: '150px',
    align: 'center',
    ellipsis: true,
    scopedSlots: {
      customRender: 'startTime'
    },
  },
  {
    title: '排查结束日期',
    dataIndex: 'endTime',
    width: '150px',
    align: 'center',
    ellipsis: true,
    scopedSlots: {
      customRender: 'endTime'
    },
  },
  {
    title: '创建时间',
    dataIndex: 'createTime',
    width: '150px',
    align: 'center',
    scopedSlots: {
      customRender: 'createTime'
    },
    ellipsis: true
  },
  {
    title: '状态',
    dataIndex: 'status',
    align: 'center',
    width: '100px',
    scopedSlots: {
      customRender: 'status'
    },
  },
  {
    title: '操作',
    dataIndex: 'action',
    align: 'center',
    width: '200px',
    scopedSlots: {
      customRender: 'action'
    },
  },
]
export const columnsPotentialFactor = [{
    title: '风险点',
    dataIndex: 'riskName',
    align: 'center',
  },
  {
    title: '管控层级',
    dataIndex: 'riskLevelName',
    align: 'center',
  },
  {
    title: '位置/区域',
    dataIndex: 'areaLocation',
    align: 'center',
  },
  {
    title: '类型',
    dataIndex: 'typeName',
    align: 'center',
  },
]
export const columnsPotentialFactorNew = [{
    title: '序号',
    customRender: (text, record, index) => `${index + 1}`,
    dataIndex: 'number',
    width: '80px',
  },
  {
    title: '风险点',
    dataIndex: 'perilPointName',
    align: 'center',
  },
  {
    title: '管控层级',
    dataIndex: 'controlLevel',
    align: 'center',
  },
  {
    title: '位置/区域',
    dataIndex: 'address',
    align: 'center',
  },
  {
    title: '类型',
    dataIndex: 'type',
    align: 'center',
  },
  {
    title: '排查人',
    dataIndex: 'checkPlanCheckPersonList',
    align: 'center',
    scopedSlots: {
      customRender: 'planCheckPerson'
    }
  },
  {
    title: '选择排查人',
    dataIndex: 'choosePlanCheckPerson',
    align: 'center',
    scopedSlots: {
      customRender: 'action'
    }
  },
]
export const columnsPotentialFactorView = [{
    title: '序号',
    customRender: (text, record, index) => `${index + 1}`,
    dataIndex: 'number',
    width: '80px',
  },

  {
    title: '风险点',
    dataIndex: 'perilPointName',
    align: 'center',
  },
  {
    title: '管控层级',
    dataIndex: 'controlLevel',
    align: 'center',
  },
  {
    title: '位置/区域',
    dataIndex: 'address',
    align: 'center',
  },
  {
    title: '类型',
    dataIndex: 'type',
    align: 'center',
  },
  {
    title: '排查人',
    dataIndex: 'checkPlanCheckPersonList',
    align: 'center',
    scopedSlots: {
      customRender: 'planCheckPerson'
    }
  }
]
