// spang-web/src/views/safety/table.js
export const importantForeColumns = [{
  title: '序号',
  customRender: (text, record, index) => `${index + 1}`,
  dataIndex: 'number',
  align: 'center',
  width: '60px'
},
{
  title: '事件描述',
  dataIndex: 'description',
  width: '350px',
  align: 'center'
},
{
  title: '报警级别',
  dataIndex: 'alarmLevel',
  width: '100px',
  align: 'center',
  scopedSlots: {
    customRender: 'alarmLevel'
  }
},
{
  title: '报警类型',
  dataIndex: 'alarmType',
  width: '100px',
  align: 'center'
},
{
  title: '报警子类型',
  dataIndex: 'alarmSubtype',
  width: '120px',
  align: 'center'
},
{
  title: '所属单位',
  dataIndex: 'enterpriseName',
  width: '190px',
  align: 'center'
},
{
  title: '发现时间',
  dataIndex: 'alarmTime',
  width: '120px',
  align: 'center'
},
{
  title: '状态',
  dataIndex: 'status',
  align: 'center',
  width: '100px',
  scopedSlots: {
    customRender: 'status'
  }
},
{
  title: '阈值',
  dataIndex: 'threshold',
  width: '120px',
  align: 'center'
}
  // {
  //   title: '是否有效',
  //   dataIndex: 'isEffective',
  //   align: 'center',
  //   width: '120px',
  //   scopedSlots: { customRender: 'isEffective' }
  // },
  // {
  //   title: '操作',
  //   dataIndex: 'action',
  //   width: '150px',
  //   align: 'center',
  //   fixed: 'right',
  //   scopedSlots: { customRender: 'action' }
  // }
]
export const forewarningColumns = [{
  title: '序号',
  customRender: (text, record, index) => `${index + 1}`,
  dataIndex: 'number',
  align: 'center',
  width: '60px'
},
{
  title: '事件描述',
  dataIndex: 'description',
  width: '350px',
  align: 'center'
},
{
  title: '报警级别',
  dataIndex: 'alarmLevel',
  width: '100px',
  align: 'center',
  scopedSlots: {
    customRender: 'alarmLevel'
  }
},
{
  title: '报警类型',
  dataIndex: 'alarmType',
  width: '100px',
  align: 'center'
},
{
  title: '报警子类型',
  dataIndex: 'alarmSubtype',
  width: '120px',
  align: 'center'
},
{
  title: '所属单位',
  dataIndex: 'enterpriseName',
  width: '190px',
  align: 'center'
},
{
  title: '发现时间',
  dataIndex: 'alarmTime',
  width: '120px',
  align: 'center'
},
{
  title: '状态',
  dataIndex: 'status',
  align: 'center',
  width: '100px',
  scopedSlots: {
    customRender: 'status'
  }
},
{
  title: '阈值',
  dataIndex: 'threshold',
  width: '120px',
  align: 'center'
},
{
  title: '是否有效',
  dataIndex: 'isEffective',
  align: 'center',
  width: '120px',
  scopedSlots: {
    customRender: 'isEffective'
  }
},
{
  title: '操作',
  dataIndex: 'action',
  width: '150px',
  align: 'center',
  fixed: 'right',
  scopedSlots: {
    customRender: 'action'
  }
}
]
// organizations
export const personnelColumns = [
  {
    title: '序号',
    dataIndex: 'serial',
    align: 'center',
    width: '80px',
    scopedSlots: {
      customRender: 'serial'
    }
  },
  {
    title: '登录账号',
    dataIndex: 'account',
    width: '120px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '姓名',
    dataIndex: 'name',
    width: '120px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '所属部门',
    dataIndex: 'deptName',
    width: '200px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '所属角色',
    dataIndex: 'roleName',
    width: '150px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '岗位',
    dataIndex: 'postName',
    width: '150px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '性别',
    dataIndex: 'sexName',
    width: '100px',
    align: 'center'
  },
  {
    title: '联系方式',
    dataIndex: 'phone',
    width: '150px',
    align: 'center',
    ellipsis: true
  }
  // {
  //   title: '详情',
  //   dataIndex: 'action',
  //   width: '150px',
  //   align: 'center',
  //   // fixed: 'right',
  //   scopedSlots: { customRender: 'action' }
  // }
]
export const organizationColumns = [{
  title: '列名1',
  dataIndex: 'column1',
  align: 'center'
},
{
  title: '列名2',
  dataIndex: 'column2',
  align: 'center'
},
{
  title: '列名3',
  dataIndex: 'column3',
  align: 'center'
},
{
  title: '操作',
  dataIndex: 'action',
  width: '150px',
  align: 'center',
  scopedSlots: {
    customRender: 'action'
  }
}
]
export const dangerColumns = [{
  title: '区域名称',
  dataIndex: 'areaName',
  width: '100px',
  align: 'center'
},
{
  title: '危险源名称',
  dataIndex: 'dangsrcName',
  width: '120px',
  align: 'center'
},
{
  title: '危险级别',
  dataIndex: 'levelName',
  width: '120px',
  align: 'center'
},
{
  title: '危险源种类',
  dataIndex: 'dangsrcTypeName',
  width: '120px',
  align: 'center'
},
{
  title: '安全距离',
  dataIndex: 'safeDistance',
  width: '120px',
  align: 'center'
},
// {
//   title: '500米内人数',
//   dataIndex: 'personNum',
//   width: '120px',
//   align: 'center'
// },
{
  title: '压力容器总数',
  dataIndex: 'pressureVesselCount',
  width: '120px',
  align: 'center'
},
{
  title: '三类压力容器总数',
  dataIndex: 'threePressureVesselCount',
  width: '120px',
  align: 'center'
},
// {
//   title: '生产经营活动类型',
//   dataIndex: 'productionType',
//   width: '120px',
//   align: 'center'
// },
// {
//   title: '生产存储场所产权',
//   dataIndex: 'placeRight',
//   width: '120px',
//   align: 'center'
// },
{
  // title: 'R值',
  dataIndex: 'rvalue',
  width: '80px',
  align: 'center',
  slots: {
    title: 'customTitle'
  },
  scopedSlots: {
    customRender: 'rvalue'
  }
},
{
  title: '详细地址',
  dataIndex: 'detailAddr',
  width: '120px',
  align: 'center'
},
// {
//   title: '经度',
//   dataIndex: 'longitude',
//   width: '120px',
//   align: 'center'
// },
// {
//   title: '纬度',
//   dataIndex: 'latitude',
//   width: '120px',
//   align: 'center'
// },
// {
//   title: '备案期限起始时间',
//   dataIndex: 'recodeStartTime',
//   width: '120px',
//   align: 'center'
// },
// {
//   title: '备案期限终止时间',
//   dataIndex: 'recodeEndTime',
//   width: '120px',
//   align: 'center'
// },
// {
//   title: '易发生事故类型',
//   dataIndex: 'accidentType',
//   width: '120px',
//   align: 'center'
// },
// {
//   title: '可能引发事故死亡人数',
//   dataIndex: 'deadNum',
//   width: '120px',
//   align: 'center'
// },
// {
//   title: '主要产品',
//   dataIndex: 'majorProduction',
//   width: '120px',
//   align: 'center'
// },
// {
//   title: '备注',
//   dataIndex: 'remark',
//   width: '120px',
//   align: 'center'
// },
{
  title: '操作',
  dataIndex: 'action',
  width: '120px',
  align: 'center',
  fixed: 'right',
  scopedSlots: {
    customRender: 'action'
  }
}
]
export const regionColumns = [{
  title: '名称',
  dataIndex: 'areaName',
  align: 'center'
},
{
  title: '创建时间',
  dataIndex: 'createTime',
  align: 'center'
},
{
  title: '是否关两重点一重大',
  dataIndex: 'isMajor',
  align: 'center'
},
{
  title: '更新时间',
  dataIndex: 'updateTime',
  align: 'center'
},
{
  title: '详情',
  dataIndex: 'action',
  width: '120px',
  align: 'center',
  // fixed: true,
  scopedSlots: {
    customRender: 'action'
  }
}
]


// orientation
export const oriPlanColumns = [{
  title: '名称',
  dataIndex: 'trainingName',
  width: '200px',
  align: 'center'
},
{
  title: '执行类型',
  dataIndex: 'executeTypeName',
  width: '120px',
  align: 'center'
},
{
  title: '计划开始时间',
  dataIndex: 'startTime',
  width: '200px',
  align: 'center'
},
{
  title: '培训主管单位',
  dataIndex: 'competentUnitName',
  width: '200px',
  align: 'center'
},
{
  title: '创建时间',
  dataIndex: 'createTime',
  width: '200px',
  align: 'center'
},
{
  title: '责任人',
  dataIndex: 'chargerName',
  width: '150px',
  align: 'center'
},
{
  title: '状态',
  dataIndex: 'statusName',
  width: '100px',
  align: 'center',
  scopedSlots: {
    customRender: 'statusName'
  }
},
{
  title: '操作',
  dataIndex: 'action',
  width: '180px',
  align: 'center',
  fixed: 'right',
  scopedSlots: {
    customRender: 'action'
  }
}
]
export const recordColumns = [{
  title: '名称',
  dataIndex: 'trainingName',
  width: '200px',
  align: 'center'
},
{
  title: '编号',
  dataIndex: 'trainingNumber',
  width: '150px',
  align: 'center'
},
{
  title: '培训形式',
  dataIndex: 'trainingType',
  width: '150px',
  align: 'center'
},
{
  title: '培训开始时间',
  dataIndex: 'startTime',
  width: '200px',
  align: 'center'
},
{
  title: '培训结束时间',
  dataIndex: 'endTime',
  width: '200px',
  align: 'center'
},
{
  title: '培训地点',
  dataIndex: 'trainingAddress',
  width: '200px',
  align: 'center'
},
{
  title: '主讲人',
  dataIndex: 'keynoteSpeaker',
  width: '150px',
  align: 'center'
},
{
  title: '学时',
  dataIndex: 'trainingPeriod',
  width: '100px',
  align: 'center'
},
// {
//   title: '是否关联计划',
//   dataIndex: 'isRelated',
//   align: 'center'
// },
{
  title: '操作',
  dataIndex: 'action',
  width: '150px',
  align: 'center',
  fixed: 'right',
  scopedSlots: {
    customRender: 'action'
  }
}
]
export const certificateColumns = [{
  title: '人员',
  dataIndex: 'userName',
  width: '150px',
  align: 'center',
  ellipsis: true
},
{
  title: '部门',
  dataIndex: 'deptName',
  width: '150px',
  align: 'center',
  ellipsis: true
},
{
  title: '证书名称',
  dataIndex: 'certName',
  width: '200px',
  align: 'center',
  ellipsis: true
},
{
  title: '证书编号',
  dataIndex: 'certCode',
  width: '210px',
  align: 'center',
  ellipsis: true
},
{
  title: '证书类型',
  dataIndex: 'workTypeName',
  width: '210px',
  align: 'center',
  ellipsis: true
},
{
  title: '发证单位',
  dataIndex: 'issuingOffice',
  width: '250px',
  align: 'center',
  ellipsis: true
},
// {
//   title: '证书认证日期',
//   dataIndex: 'approveTime',
//   width: '200px',
//   align: 'center'
// },
// {
//   title: '证书到期日期',
//   dataIndex: 'expireTime',
//   width: '200px',
//   align: 'center'
// },
// {
//   title: '原件存放',
//   dataIndex: 'storeAddress',
//   width: '150px',
//   align: 'center'
// },
// {
//   title: '认证项目',
//   dataIndex: 'approveProject',
//   width: '200px',
//   align: 'center'
// },
// {
//   title: '创建人',
//   dataIndex: 'createUserName',
//   width: '200px',
//   align: 'center'
// },
{
  title: '操作',
  dataIndex: 'action',
  width: '280px',
  align: 'center',
  // fixed: 'right',
  scopedSlots: {
    customRender: 'action'
  }
}
]
export const educationColumns = [{
  title: '员工',
  dataIndex: 'personName',
  align: 'center'
},
{
  title: '培训课时',
  dataIndex: 'trainingPeriod',
  align: 'center'
},
{
  title: '授课开始日期',
  dataIndex: 'startTime',
  align: 'center'
},
{
  title: '考试成绩',
  dataIndex: 'score',
  align: 'center'
},
{
  title: '教育类型',
  dataIndex: 'educateTypeName',
  align: 'center'
},
{
  title: '创建时间',
  dataIndex: 'createTime',
  align: 'center'
},
{
  title: '创建人',
  dataIndex: 'createUsername',
  align: 'center'
},
{
  title: '操作',
  dataIndex: 'action',
  width: '150px',
  align: 'center',
  scopedSlots: {
    customRender: 'action'
  }
}
]
export const personEduColumns = [{
  title: '序号',
  width: '150px',
  scopedSlots: {
    customRender: 'serial'
  }
},
{
  title: '员工',
  dataIndex: 'personName',
  width: '150px',
  align: 'center'
},
{
  title: '部门',
  dataIndex: 'deptName',
  width: '150px',
  align: 'center'
},
{
  title: '岗位',
  dataIndex: 'deptName',
  width: '150px',
  align: 'center'
},
{
  title: '创建时间',
  width: '150px',
  dataIndex: 'creatdTime',
  align: 'center'
},
{
  title: '更新时间',
  width: '150px',
  dataIndex: 'updatedTime',
  align: 'center'
},
{
  title: '三级安全教育',
  dataIndex: 'education',
  width: '150px',
  align: 'center',
  // fixed: 'right',
  scopedSlots: {
    customRender: 'education'
  }
},
{
  title: '特殊工种证书',
  dataIndex: 'certificate',
  width: '150px',
  align: 'center',
  // fixed: 'right',
  scopedSlots: {
    customRender: 'certificate'
  }
}
]

// risk
export const riskImportant = [
  {
    title: '序号',
    width: '80px',
    scopedSlots: {
      customRender: 'serial'
    }
  },
  {
    title: '名称',
    dataIndex: 'mainName',
    align: 'center',
    ellipsis: true,
    width: '200px'
  },
  {
    title: '级别',
    dataIndex: 'levelName',
    width: '80px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '类型',
    dataIndex: 'typeName',
    align: 'center',
    width: '200px',
    ellipsis: true
  },
  {
    title: '安全责任人',
    dataIndex: 'dutyUserName',
    width: '180px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '200px',
    align: 'center',
    // fixed: 'right',
    scopedSlots: {
      customRender: 'action'
    }
  }
]

export const hazopColumns = [{
  title: '序号',
  scopedSlots: {
    customRender: 'serial'
  },
  dataIndex: 'index',
  width: '80px',
  align: 'center',
  ellipsis: true
},
{
  title: '报告名称',
  dataIndex: 'reportName',
  width: '200px',
  align: 'center',
  ellipsis: true
},
{
  title: '评估机构',
  dataIndex: 'assessOrgName',
  width: '180px',
  align: 'center',
  ellipsis: true
},
{
  title: '参与单位',
  dataIndex: 'participatoryUnit',
  width: '180px',
  align: 'center',
  ellipsis: true
},
{
  title: '报告日期',
  dataIndex: 'reportDate',
  width: '120px',
  align: 'center',
  ellipsis: true
},
{
  title: '状态',
  dataIndex: 'status',
  width: '80px',
  align: 'center',
  scopedSlots: {
    customRender: 'status'
  }
  // customRender: (text) => {
  //   return text ? '已发布' : '未发布'
  // }
},
{
  title: '操作',
  dataIndex: 'action',
  width: '200px',
  align: 'center',
  // fixed: 'right',
  scopedSlots: {
    customRender: 'action'
  }
}
]

// areaColumns
export const areaColumns = [
  {
    title: '序号',
    scopedSlots: {
      customRender: 'serial'
    },
    dataIndex: 'index',
    width: '80px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '区域/位置',
    dataIndex: 'areaLocation',
    width: '200px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '单位/部门',
    dataIndex: 'deptName',
    width: '200px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '创建人',
    dataIndex: 'gmtCreateName',
    align: 'center',
    width: '120px',
    ellipsis: true
  },
  {
    title: '区域编号',
    dataIndex: 'areaCode',
    align: 'center',
    width: '150px',
    ellipsis: true
  },
  {
    title: '创建时间',
    dataIndex: 'gmtCreate',
    width: '150px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '200px',
    align: 'center',
    scopedSlots: {
      customRender: 'action'
    }
  }
]

// 风险管控层级
export const riskLevelColumns = [{
  title: '序号',
  customRender: (text, record, index) => `${index + 1}`,
  dataIndex: 'index',
  width: '80px',
  align: 'center'
},
{
  title: '风险等级',
  dataIndex: 'riskLevelName',
  align: 'center'
},
{
  title: '风险色',
  dataIndex: 'riskColourName',
  align: 'center'
},
{
  title: '风险控制层级',
  dataIndex: 'riskControlName',
  align: 'center'
},
{
  title: '操作',
  dataIndex: 'action',
  width: '200px',
  align: 'center',
  fixed: 'right',
  scopedSlots: {
    customRender: 'action'
  }
}
]

export const workRiskColumns = [{
  title: '所属单位',
  dataIndex: 'unitName',
  width: '200px',
  align: 'center'
},
{
  title: '所属区域',
  dataIndex: 'areaName',
  width: '120px',
  align: 'center'
},
{
  title: '作业活动',
  dataIndex: 'work',
  width: '200px',
  align: 'center'
},
{
  title: '主要后果',
  dataIndex: 'majorOutcome',
  width: '120px',
  align: 'center'
},
{
  // title: 'R',
  dataIndex: 'r',
  align: 'center',
  width: '80px',
  slots: {
    title: 'customTitle'
  },
  scopedSlots: {
    customRender: 'r'
  }
},
{
  title: '评价级别',
  dataIndex: 'evaluationLevelName',
  width: '150px',
  align: 'center'
},
{
  title: '管控级别',
  dataIndex: 'controlLevelName',
  width: '150px',
  align: 'center'
},
{
  title: '责任单位',
  dataIndex: 'chargerUnitName',
  width: '200px',
  align: 'center'
},
{
  title: '责任人',
  dataIndex: 'chargerName',
  width: '150px',
  align: 'center'
},
// {
//   title: '辨识时间',
//   dataIndex: 'findingTime',
//   width: '200px',
//   align: 'center'
// },
{
  title: '操作',
  dataIndex: 'action',
  width: '120px',
  align: 'center',
  fixed: 'right',
  scopedSlots: {
    customRender: 'action'
  }
}
]

export const technologyColumns = [{
  title: '所属单位',
  dataIndex: 'unitName',
  width: '200px',
  align: 'center'
},
{
  title: '所属区域',
  dataIndex: 'areaName',
  width: '150px',
  align: 'center'
},
{
  title: '工艺过程',
  dataIndex: 'process',
  width: '150px',
  align: 'center'
},
{
  title: '主要后果',
  dataIndex: 'majorOutcome',
  width: '150px',
  align: 'center'
},
{
  title: 'R',
  dataIndex: 'r',
  width: '80px',
  align: 'center'
},
{
  title: '评价级别',
  dataIndex: 'evaluationLevelName',
  width: '150px',
  align: 'center'
},
{
  title: '管控级别',
  dataIndex: 'controlLevelName',
  width: '150px',
  align: 'center'
},
{
  title: '责任单位',
  dataIndex: 'chargerUnitName',
  width: '200px',
  align: 'center'
},
// {
//   title: '责任人',
//   dataIndex: 'chargerName',
//   width: '150px',
//   align: 'center'
// },
// {
//   title: '辨识时间',
//   dataIndex: 'findingTime',
//   width: '200px',
//   align: 'center'
// },
{
  title: '操作',
  dataIndex: 'action',
  width: '120px',
  align: 'center',
  fixed: 'right',
  scopedSlots: {
    customRender: 'action'
  }
}
]



//风险评价方法设置
export const riskEvaluationColumns1 = [{
  title: '分数',
  dataIndex: 'scores',
  align: 'center',
  ellipsis: true,
  width: '60px',
},
{
  title: '事件发生频率',
  dataIndex: 'happenFrequency',
  align: 'center',
  ellipsis: true,
  width: '120px',
},
{
  title: '安全检查',
  dataIndex: 'securityCheck',
  align: 'center',
  ellipsis: true,
  width: '100px',
},
{
  title: '操作规程',
  dataIndex: 'workRule',
  align: 'center',
  ellipsis: true,
  width: '100px',
},
{
  title: '员工胜任程度（意识、技能、经验）',
  dataIndex: 'employeeCompetency',
  align: 'center',
  ellipsis: true,
  width: '260px',
},
{
  title: '防范、控制措施',
  dataIndex: 'controlMeasures',
  align: 'center',
  ellipsis: true,
},
{
  title: '操作',
  dataIndex: 'action',
  width: '200px',
  align: 'center',
  fixed: 'right',
  scopedSlots: {
    customRender: 'action'
  }
}
]
export const riskEvaluationColumns2 = [{
  title: '分数',
  dataIndex: 'scores',
  align: 'center',
  ellipsis: true,
  width: '60px',
},
{
  title: '人员伤亡程度',
  dataIndex: 'casualties',
  align: 'center',
  ellipsis: true,
  width: '120px',
},
{
  title: '直接财产损失',
  dataIndex: 'propertyLoss',
  align: 'center',
  ellipsis: true,
  width: '120px',
},
{
  title: '停工时间',
  dataIndex: 'downtime',
  align: 'center',
  ellipsis: true,
  width: '100px',
},
{
  title: '对声誉的影响',
  dataIndex: 'reputationImpact',
  align: 'center',
  ellipsis: true,
  width: '120px',
},
{
  title: '操作',
  dataIndex: 'action',
  width: '200px',
  align: 'center',
  fixed: 'right',
  scopedSlots: {
    customRender: 'action'
  }
}
]
export const riskEvaluationColumns3 = [{
  title: '风险度',
  dataIndex: 'judge',
  customRender: (text, record) => (record.judgeMin + '-' + record.judgeMax),
  align: 'center',
  ellipsis: true,
  width: '100px',
},
{
  title: '风险等级',
  dataIndex: 'riskLevelName',
  align: 'center',
  ellipsis: true,
  width: '100px',
},
{
  title: '应采取的行动/控制措施',
  dataIndex: 'controlMeasures',
  align: 'center',
  ellipsis: true,
  width: '250px',
},
{
  title: '实施期限',
  dataIndex: 'timeLimit',
  align: 'center',
  ellipsis: true,
  width: '100px',
},
{
  title: '操作',
  dataIndex: 'action',
  width: '200px',
  align: 'center',
  fixed: 'right',
  scopedSlots: {
    customRender: 'action'
  }
}
]
export const riskEvaluationColumns4 = [{
  title: '风险级别',
  dataIndex: 'riskLevelName',
  align: 'center',
  ellipsis: true,
  width: '100px',
},
{
  title: '风险颜色',
  dataIndex: 'riskColourName',
  align: 'center',
  ellipsis: true,
  width: '100px',
},
{
  title: '风险值（起始值-结束值）',
  dataIndex: 'riskScore',
  customRender: (text, record) => (record.riskBgScore + '-' + record.riskEndScore),
  align: 'center',
  ellipsis: true,
  width: '250px',
},
{
  title: '操作',
  dataIndex: 'action',
  width: '200px',
  align: 'center',
  fixed: 'right',
  scopedSlots: {
    customRender: 'action'
  }
}
]
// law
export const lawColumns = [{
  title: '序号',
  align: 'center',
  width: '60px',
  scopedSlots: {
    customRender: 'number'
  }
},
{
  title: '文件名称',
  dataIndex: 'documentName',
  align: 'center',
  ellipsis: true,
  width: '100px',
},
{
  title: '文号',
  dataIndex: 'documentCode',
  align: 'center',
  ellipsis: true,
  width: '100px',
},
{
  title: '发文单位',
  dataIndex: 'releaseOrgName',
  align: 'center',
  ellipsis: true,
  width: '100px',
},
{
  title: '文件有效性',
  dataIndex: 'documentValidName',
  align: 'center',
  ellipsis: true,
  width: '130px',
},
{
  title: '文件类别',
  dataIndex: 'documentTypeName',
  align: 'center',
  ellipsis: true,
  width: '100px',
},
{
  title: '颁布日期',
  dataIndex: 'issueDate',
  align: 'center',
  ellipsis: true,
  width: '100px',
},
{
  title: '实施日期',
  dataIndex: 'implementationDate',
  align: 'center',
  ellipsis: true,
  width: '100px',
},
{
  title: '符合性',
  dataIndex: 'selfFitName',
  align: 'center',
  ellipsis: true,
  width: '100px',
},
{
  title: '适用性',
  dataIndex: 'selfApplyName',
  align: 'center',
  ellipsis: true,
  width: '100px',
},
{
  title: '创建人',
  dataIndex: 'gmtCreateName',
  align: 'center',
  ellipsis: true,
  width: '100px',
},
{
  title: '状态',
  dataIndex: 'status',
  scopedSlots: {
    customRender: 'status'
  },
  align: 'center',
  ellipsis: true,
  width: '100px',
},
{
  title: '操作',
  dataIndex: 'action',
  width: '250px',
  align: 'center',
  // fixed: 'right',
  scopedSlots: {
    customRender: 'action'
  }
}
]
// rules
export const rulesColumns = [{
  title: '序号',
  align: 'center',
  width: '60px',
  scopedSlots: {
    customRender: 'number'
  }
}, {
  title: '文件编号',
  dataIndex: 'documentCode',
  align: 'center',
  ellipsis: true,
  width: '150px',
},
{
  title: '文件名称',
  dataIndex: 'documentName',
  align: 'center',
  ellipsis: true,
  width: '150px',
},
{
  title: '文件类别',
  dataIndex: 'documentTypeName',
  align: 'center',
  ellipsis: true,
  width: '150px',
},
{
  title: '文件有效性',
  dataIndex: 'documentValidType',
  align: 'center',
  ellipsis: true,
  width: '150px',
},
{
  title: '发布日期',
  dataIndex: 'releaseDate',
  align: 'center',
  ellipsis: true,
  width: '150px',
},
{
  title: '创建日期',
  dataIndex: 'gmtCreate',
  align: 'center',
  ellipsis: true,
  width: '150px',
},
{
  title: '创建人',
  dataIndex: 'gmtCreateName',
  align: 'center',
  ellipsis: true,
  width: '100px',
},
{
  title: '状态',
  dataIndex: 'status',
  ellipsis: true,
  width: '100px',
  align: 'center',
  scopedSlots: {
    customRender: 'status'
  }
},
{
  title: '操作',
  dataIndex: 'action',
  width: '250px',
  align: 'center',
  // fixed: 'right',
  scopedSlots: {
    customRender: 'action'
  }
}
]
//commitmentModelPerson
export const commitmentModelPersonColumns = [{
  title: '序号',
  align: 'center',
  width: '60px',
  scopedSlots: {
    customRender: 'number'
  }
},
{
  title: '类型',
  dataIndex: 'templateTypeName',
  align: 'center',
  width: '100px',
  ellipsis: true
},
{
  title: '名称',
  dataIndex: 'templateName',
  align: 'center',
  width: '200px',
  ellipsis: true
},
{
  title: '签字单位',
  dataIndex: 'signOrgName',
  align: 'center',
  width: '200px',
  ellipsis: true
},
{
  title: '岗位',
  dataIndex: 'postName',
  align: 'center',
  width: '200px',
  ellipsis: true,
},
{
  title: '签字人',
  dataIndex: 'userName',
  align: 'center',
  width: '150px',
  ellipsis: true,
},
{
  title: '签字日期',
  dataIndex: 'userSignDate',
  width: '150px',
  ellipsis: true,
  align: 'center'
},
{
  title: '状态',
  dataIndex: 'userIsSign',
  align: 'center',
  width: '100px',
  ellipsis: true,
  scopedSlots: {
    customRender: 'userIsSign'
  }
},
{
  title: '操作',
  dataIndex: 'action',
  width: '200px',
  align: 'center',
  // fixed: 'right',
  scopedSlots: {
    customRender: 'action'
  }
}
]
//targetResponsibility
export const targetResponsibilityColumns = [

  {
    title: '序号',
    align: 'center',
    width: '60px',
    scopedSlots: {
      customRender: 'number'
    }
  },
  {
    title: '模板类型',
    dataIndex: 'templateTypeName',
    width: '200px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '模板名称',
    dataIndex: 'templateName',
    width: '150px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '签字单位',
    dataIndex: 'signOrgName',
    width: '150px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '岗位',
    dataIndex: 'postName',
    width: '100px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '创建人',
    dataIndex: 'gmtCreateName',
    width: '100px',
    align: 'center'
  },
  {
    title: '创建时间',
    dataIndex: 'gmtCreate',
    width: '200px',
    align: 'center'
  },
  {
    title: '发布时间',
    dataIndex: 'pubDate',
    width: '100px',
    align: 'center'
  },
  {
    title: '状态',
    dataIndex: 'status',
    scopedSlots: {
      customRender: 'status'
    },
    width: '80px',
    align: 'center'
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '200px',
    align: 'center',
    // fixed: 'right',
    scopedSlots: {
      customRender: 'action'
    }
  }
]
//maintenance
export const maintenanceColumns = [

  {
    title: '序号',
    align: 'center',
    width: '60px',
    scopedSlots: {
      customRender: 'number'
    }
  },
  {
    title: '日期',
    dataIndex: 'madeDate',
    width: '100px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '状态',
    dataIndex: 'status',
    width: '150px',
    align: 'center',
    ellipsis: true,
    scopedSlots: {
      customRender: 'status'
    },
    customRender: text => {
      if (text == 0) {
        return '已审批'
      }
      if (text == 1) {
        return '草稿'
      }
      if (text == 2) {
        return '提交'
      }
    }

  },
  {
    title: '审批人',
    dataIndex: 'approveName',
    width: '150px',
    align: 'center',
    ellipsis: true,
  },

  {
    title: '审批时间',
    dataIndex: 'approveTime',
    width: '200px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '提交时间',
    dataIndex: 'submitTime',
    width: '150px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '创建人',
    dataIndex: 'gmtCreateName',
    width: '100px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '创建时间',
    dataIndex: 'gmtCreate',
    width: '150px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '200px',
    align: 'center',
    // fixed: 'right',
    scopedSlots: {
      customRender: 'action'
    }
  }
]
// orientationplan
export const orientationplan = [
  //   {
  //   title: '序号',
  //   align: 'center',
  //   width: '60px',
  //   scopedSlots: {
  //     customRender: 'number'
  //   }
  // },
  {
    title: '名称',
    dataIndex: 'trainName',
    ellipsis: true,
    width: '200px',
    align: 'center'
  },
  {
    title: '培训类型',
    dataIndex: 'trainTypeName',
    ellipsis: true,
    width: '100px',
    align: 'center'
  },
  {
    title: '培训级别',
    dataIndex: 'trainLevelName',
    ellipsis: true,
    width: '100px',
    align: 'center'
  },
  {
    title: '培训组织单位',
    dataIndex: 'orgUnitName',
    ellipsis: true,
    width: '150px',
    align: 'center'
  },
  {
    title: '培训日期',
    dataIndex: '',
    customRender: (text, record) => (record.trainStartDate + '-' + record.trainEndDate),
    width: '300px',
    align: 'center'
  },
  {
    title: '创建人',
    dataIndex: 'createName',
    ellipsis: true,
    width: '150px',
    align: 'center'
  },
  {
    title: '状态',
    dataIndex: 'planStatus',
    scopedSlots: {
      customRender: 'planStatus'
    },
    // customRender: (text) => {
    //   if (text == 0) {
    //     return '未提交'
    //   }
    //   if (text == 1) {
    //     return '待审批'
    //   }
    //   if (text == 2) {
    //     return '已审批'
    //   }
    //   if (text == 3) {
    //     return '已驳回'
    //   }
    // },
    width: '200px',
    align: 'center'
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '240px',
    align: 'center',
    // fixed: 'right',
    scopedSlots: {
      customRender: 'action'
    }
  }
]
//examination
export const examination = [{
  title: '序号',
  align: 'center',
  width: '60px',
  scopedSlots: {
    customRender: 'number'
  }
}, {
  title: '名称',
  dataIndex: 'trainName',
  ellipsis: true,
  width: '150px',
  align: 'center'
},
{
  title: '培训类型',
  dataIndex: 'trainTypeName',
  ellipsis: true,
  width: '100px',
  align: 'center'
},
{
  title: '培训级别',
  dataIndex: 'trainLevelName',
  ellipsis: true,
  width: '150px',
  align: 'center'
},
{
  title: '培训组织单位',
  dataIndex: 'orgUnitName',
  ellipsis: true,
  width: '150px',
  align: 'center'
},
{
  title: '培训日期',
  dataIndex: '',
  ellipsis: true,
  customRender: (text, record) => (record.trainStartDate + '-' + record.trainEndDate),
  width: '300px',
  align: 'center'
},
{
  title: '创建人',
  dataIndex: 'createName',
  ellipsis: true,
  width: '150px',
  align: 'center'
},
{
  title: '状态',
  dataIndex: 'planStatus',
  scopedSlots: {
    customRender: 'planStatus'
  },
  // customRender: (text) => {
  //   if (text == 1) {
  //     return '待审批'
  //   }
  // },
  width: '200px',
  align: 'center'
},
{
  title: '操作',
  dataIndex: 'action',
  width: '240px',
  align: 'center',
  // fixed: 'right',
  scopedSlots: {
    customRender: 'action'
  }
}
]
//3registry
export const registryColumns = [{
  title: '序号',
  align: 'center',
  width: '60px',
  scopedSlots: {
    customRender: 'number'
  }
}, {
  title: '三违行为类型',
  dataIndex: 'violationTypeName',
  align: 'center',
  ellipsis: true,
  width: '150px',
},
{
  title: '描述',
  dataIndex: 'description',
  align: 'center',
  ellipsis: true,
  width: '150px',
},
{
  title: '违规人姓名',
  dataIndex: 'violationPersonName',
  align: 'center',
  ellipsis: true,
  width: '150px',
},
{
  title: '违规人部门',
  dataIndex: 'violationDeptName',
  align: 'center',
  ellipsis: true,
  width: '150px',
},
{
  title: '登记人',
  dataIndex: 'registrantName',
  align: 'center',
  ellipsis: true,
  width: '150px',
},
{
  title: '登记时间',
  dataIndex: 'registerTime',
  align: 'center',
  ellipsis: true,
  width: '150px',
},
{
  title: '附件',
  dataIndex: 'files',
  align: 'center',
  ellipsis: true,
  width: '100px',
  scopedSlots: {
    customRender: 'files'
  }
},
{
  title: '状态',
  dataIndex: 'status',
  ellipsis: true,
  width: '100px',
  align: 'center',
  scopedSlots: {
    customRender: 'status'
  },
  // customRender: text => {
  //   if (text == 0) {
  //     return '未提交'
  //   }
  //   if (text == 1) {
  //     return '已提交'
  //   }
  //   if (text == 2) {
  //     return '已处理'
  //   }
  //   if (text == 3) {
  //     return '关闭'
  //   }
  // }
},
{
  title: '操作',
  dataIndex: 'action',
  width: '250px',
  align: 'center',
  // fixed: 'right',
  scopedSlots: {
    customRender: 'action'
  }
}
]
//deal
export const dealColumns = [{
  title: '序号',
  align: 'center',
  width: '60px',
  scopedSlots: {
    customRender: 'number'
  }
}, {
  title: '三违行为类型',
  dataIndex: 'violationTypeName',
  align: 'center',
  ellipsis: true,
  width: '150px',
},
{
  title: '描述',
  dataIndex: 'description',
  align: 'center',
  ellipsis: true,
  width: '150px',
},
{
  title: '违规人姓名',
  dataIndex: 'violationPersonName',
  align: 'center',
  ellipsis: true,
  width: '150px',
},
{
  title: '违规人部门',
  dataIndex: 'violationDeptName',
  align: 'center',
  ellipsis: true,
  width: '150px',
},
{
  title: '登记人',
  dataIndex: 'registrantName',
  align: 'center',
  ellipsis: true,
  width: '150px',
},
{
  title: '登记时间',
  dataIndex: 'registerTime',
  align: 'center',
  ellipsis: true,
  width: '150px',
},
{
  title: '附件',
  dataIndex: 'files',
  align: 'center',
  ellipsis: true,
  width: '100px',
  scopedSlots: {
    customRender: 'files'
  }
},
{
  title: '处理人',
  dataIndex: 'processorName',
  align: 'center',
  ellipsis: true,
  // width: '150px',
},
{
  title: '状态',
  dataIndex: 'status',
  ellipsis: true,
  width: '100px',
  align: 'center',
  scopedSlots: {
    customRender: 'status'
  }
},
{
  title: '操作',
  dataIndex: 'action',
  width: '250px',
  align: 'center',
  // fixed: 'right',
  scopedSlots: {
    customRender: 'action'
  }
}
]
//chemicals
export const chemicalsColumns = [{
  title: '序号',
  align: 'center',
  width: '60px',
  scopedSlots: {
    customRender: 'number'
  }
}, {
  title: '危化品名称',
  dataIndex: 'chemicalName',
  align: 'center',
  ellipsis: true,
  width: '150px',
},
{
  title: '存储方式',
  dataIndex: 'storageMode',
  align: 'center',
  ellipsis: true,
  width: '150px',
},
{
  title: '存储地点',
  dataIndex: 'storageAddress',
  align: 'center',
  ellipsis: true,
  width: '150px',
},
{
  title: '状态',
  dataIndex: 'status',
  ellipsis: true,
  width: '100px',
  align: 'center',
  scopedSlots: {
    customRender: 'status'
  }
},
{
  title: '操作',
  dataIndex: 'action',
  width: '250px',
  align: 'center',
  // fixed: 'right',
  scopedSlots: {
    customRender: 'action'
  }
}
]
//costPlan
export const costPlanColumns = [{
  title: '序号',
  align: 'center',
  width: '60px',
  scopedSlots: {
    customRender: 'number'
  }
}, {
  title: '年度',
  dataIndex: 'year',
  align: 'center',
  ellipsis: true,
  width: '150px',
},
{
  title: '安全生产费用（元）',
  dataIndex: 'safetyProductionCost',
  align: 'center',
  ellipsis: true,
  width: '200px',
},
{
  title: '月度计提标准（元）',
  dataIndex: 'monthlyProvisionStandard',
  align: 'center',
  ellipsis: true,
  width: '200px',
},
{
  title: '操作',
  dataIndex: 'action',
  width: '250px',
  align: 'center',
  // fixed: 'right',
  scopedSlots: {
    customRender: 'action'
  }
}
]
//costReceive
export const costReceiveColumns = [{
  title: '序号',
  align: 'center',
  width: '60px',
  scopedSlots: {
    customRender: 'number'
  }
}, {
  title: '类型',
  dataIndex: 'typeName',
  align: 'center',
  ellipsis: true,
  width: '150px',
},
{
  title: '领用单位',
  dataIndex: 'recipientsUnitName',
  align: 'center',
  ellipsis: true,
  width: '150px',
},
{
  title: '使用项目',
  dataIndex: 'useProjectName',
  align: 'center',
  ellipsis: true,
  width: '150px',
}, {
  title: '费用',
  dataIndex: 'cost',
  align: 'center',
  ellipsis: true,
  width: '150px',
},
{
  title: '领用日期',
  dataIndex: 'receivingDate',
  align: 'center',
  ellipsis: true,
  width: '150px',
},
{
  title: '领用人',
  dataIndex: 'recorderName',
  align: 'center',
  ellipsis: true,
  width: '150px',
},
{
  title: '状态',
  dataIndex: 'status',
  ellipsis: true,
  width: '100px',
  align: 'center',
  scopedSlots: {
    customRender: 'status'
  }
},
{
  title: '操作',
  dataIndex: 'action',
  width: '250px',
  align: 'center',
  // fixed: 'right',
  scopedSlots: {
    customRender: 'action'
  }
}
]
//costStatistic
export const costStatisticColumns = [{
  title: '序号',
  align: 'center',
  width: '60px',
  scopedSlots: {
    customRender: 'number'
  }
}, {
  title: '使用项目',
  dataIndex: 'typeName',
  align: 'center',
  ellipsis: true,
  width: '150px',
},
{
  title: '使用费用（元）',
  dataIndex: 'sum',
  align: 'center',
  ellipsis: true,
  width: '150px',
},
{
  title: '百分比',
  dataIndex: 'percentage',
  align: 'center',
  ellipsis: true,
  width: '150px',
}
]
export const costStatisticTableColumns = [{
  title: '使用项目',
  dataIndex: 'typeName',
  align: 'center',
  ellipsis: true,
  width: '150px',
},
{
  title: '1月',
  dataIndex: '安全防护设施费用[0]',
  align: 'center',
  ellipsis: true,
  width: '100px',
},
{
  title: '2月',
  dataIndex: 'recipientsUnitName',
  align: 'center',
  ellipsis: true,
  width: '100px',
},
{
  title: '3月',
  dataIndex: 'recipientsUnitName',
  align: 'center',
  ellipsis: true,
  width: '100px',
},
{
  title: '4月',
  dataIndex: 'recipientsUnitName',
  align: 'center',
  ellipsis: true,
  width: '100px',
},
{
  title: '5月',
  dataIndex: 'recipientsUnitName',
  align: 'center',
  ellipsis: true,
  width: '100px',
},
{
  title: '6月',
  dataIndex: 'recipientsUnitName',
  align: 'center',
  ellipsis: true,
  width: '100px',
},
{
  title: '7月',
  dataIndex: 'recipientsUnitName',
  align: 'center',
  ellipsis: true,
  width: '100px',
},
{
  title: '8月',
  dataIndex: 'recipientsUnitName',
  align: 'center',
  ellipsis: true,
  width: '100px',
},
{
  title: '9月',
  dataIndex: 'recipientsUnitName',
  align: 'center',
  ellipsis: true,
  width: '100px',
},
{
  title: '10月',
  dataIndex: 'recipientsUnitName',
  align: 'center',
  ellipsis: true,
  width: '100px',
},
{
  title: '11月',
  dataIndex: 'recipientsUnitName',
  align: 'center',
  ellipsis: true,
  width: '100px',
},
{
  title: '12月',
  dataIndex: 'recipientsUnitName',
  align: 'center',
  ellipsis: true,
  width: '100px',
},
]



// schedule
export const workPlanColumns = [{
  title: '名称',
  dataIndex: 'workPlanName',
  width: '150px',
  align: 'center'
},
{
  title: '工作内容',
  dataIndex: 'workContent',
  width: '200px',
  align: 'center'
},
{
  title: '工作类型',
  dataIndex: 'workTypeName',
  width: '120px',
  align: 'center'
},
{
  title: '执行周期',
  dataIndex: 'executePeriodName',
  width: '120px',
  align: 'center'
},
{
  title: '工作周期天数',
  dataIndex: 'workDays',
  width: '100px',
  align: 'center'
},
{
  title: '培训提醒天数',
  dataIndex: 'remindDays',
  width: '100px',
  align: 'center'
},
{
  title: '计划开始时间',
  dataIndex: 'startTime',
  width: '150px',
  align: 'center'
},
{
  title: '创建时间',
  dataIndex: 'createTime',
  width: '150px',
  align: 'center'
},
{
  title: '创建人',
  dataIndex: 'createUserName',
  width: '150px',
  align: 'center'
},
{
  title: '状态',
  dataIndex: 'statusName',
  width: '100px',
  align: 'center',
  scopedSlots: {
    customRender: 'statusName'
  }
},
{
  title: '操作',
  dataIndex: 'action',
  width: '150px',
  align: 'center',
  scopedSlots: {
    customRender: 'action'
  }
}
]
// workPlan todo
const prefix = 'workPlanVO.'
export const workPlanTodoColumns = [{
  title: '名称',
  dataIndex: prefix + 'workPlanName',
  width: '150px',
  align: 'center'
},
{
  title: '工作类型',
  dataIndex: prefix + 'workTypeName',
  width: '150px',
  align: 'center'
},
{
  title: '当前计划时间',
  dataIndex: prefix + 'startTime',
  width: '150px',
  align: 'center'
},
{
  title: '责任单位',
  dataIndex: prefix + 'dutyUnitName',
  width: '150px',
  align: 'center'
},
{
  title: '责任人',
  dataIndex: prefix + 'chargerName',
  width: '150px',
  align: 'center'
},
{
  title: '创建时间',
  dataIndex: prefix + 'createTime',
  width: '150px',
  align: 'center'
},
{
  title: '创建人',
  dataIndex: prefix + 'createUserName',
  width: '150px',
  align: 'center'
},
{
  title: '状态',
  dataIndex: prefix + 'statusName',
  width: '150px',
  align: 'center',
  scopedSlots: {
    customRender: 'statusName'
  }
},
{
  title: '操作',
  dataIndex: 'action',
  width: '450px',
  align: 'center',
  scopedSlots: {
    customRender: 'action'
  }
}
]
export const workRecordColumns = [{
  title: '名称',
  dataIndex: 'workName',
  width: '150px',
  align: 'center'
},
{
  title: '工作类型',
  dataIndex: 'workTypeName',
  width: '120px',
  align: 'center'
},
{
  title: '完成时间',
  dataIndex: 'endTime',
  width: '200px',
  align: 'center'
},
{
  title: '责任单位',
  dataIndex: 'dutyUnitName',
  width: '200px',
  align: 'center'
},
{
  title: '责任人',
  dataIndex: 'chargerName',
  width: '150px',
  align: 'center'
},
{
  title: '创建时间',
  dataIndex: 'createTime',
  width: '200px',
  align: 'center'
},
{
  title: '创建人',
  dataIndex: 'createUserName',
  width: '150px',
  align: 'center'
},
// {
//   title: '是否关联工作计划',
//   dataIndex: 'isRelated',
//   align: 'center'
// },
{
  title: '操作',
  dataIndex: 'action',
  width: '120px',
  align: 'center',
  scopedSlots: {
    customRender: 'action'
  }
}
]
export const inspectionPlanColumns = [{
  title: '名称',
  dataIndex: 'examinePlanName',
  width: '200px',
  align: 'center'
},
{
  title: '检查类型',
  dataIndex: 'examineTypeName',
  width: '150px',
  align: 'center'
},
{
  title: '当前计划时间',
  dataIndex: 'startTime',
  width: '200px',
  align: 'center'
},
{
  title: '责任单位',
  dataIndex: 'dutyUnitName',
  width: '180px',
  align: 'center'
},
{
  title: '责任人',
  dataIndex: 'chargerName',
  width: '150px',
  align: 'center'
},
{
  title: '创建时间',
  dataIndex: 'createTime',
  width: '200px',
  align: 'center'
},
{
  title: '创建人',
  dataIndex: 'createUserName',
  width: '150px',
  align: 'center'
},
{
  title: '状态',
  dataIndex: 'stautusName',
  width: '150px',
  align: 'center'
},
{
  title: '操作',
  dataIndex: 'action',
  width: '120px',
  align: 'center',
  fixed: 'right',
  scopedSlots: {
    customRender: 'action'
  }
}
]
export const inspectionRecordColumns = [{
  title: '名称',
  dataIndex: 'examinePlanName',
  width: '200px',
  align: 'center'
},
{
  title: '检查类型',
  dataIndex: 'examineTypeName',
  width: '120px',
  align: 'center'
},
{
  title: '完成时间',
  dataIndex: 'endTime',
  width: '200px',
  align: 'center'
},
{
  title: '责任单位',
  dataIndex: 'dutyUnitName',
  width: '200px',
  align: 'center'
},
{
  title: '责任人',
  dataIndex: 'chargerName',
  width: '150px',
  align: 'center'
},
{
  title: '创建时间',
  dataIndex: 'createTime',
  width: '200px',
  align: 'center'
},
{
  title: '创建人',
  dataIndex: 'createUserName',
  width: '150px',
  align: 'center'
},
// {
//   title: '是否关联检查计划',
//   dataIndex: 'isRelated',
//   align: 'center'
// },
{
  title: '操作',
  dataIndex: 'action',
  width: '120px',
  align: 'center',
  scopedSlots: {
    customRender: 'action'
  }
}
]

// potential risk
export const reformColumns = [
  // {
  //   title: '责任单位',
  //   dataIndex: 'dutyUnitName',
  //   align: 'center',
  //   width: '150px'
  // },
  {
    title: '隐患描述',
    dataIndex: 'perilDescription',
    align: 'center',
    width: '300px'
  },
  {
    title: '填报人',
    dataIndex: 'informantName',
    width: '150px',
    align: 'center'
  },
  {
    title: '隐患类型',
    dataIndex: 'typeName',
    align: 'center',
    width: '200px'
  },
  {
    title: '整改期限',
    dataIndex: 'abarbeitungDeadline',
    width: '150px',
    align: 'center'
  },
  // {
  //   title: '整改责任人',
  //   dataIndex: 'dutyPersonName',
  //   align: 'center',
  //   width: '150px'
  // },
  {
    title: '检查时间',
    dataIndex: 'checkTime',
    align: 'center',
    width: '150px'
  },
  {
    title: '状态',
    dataIndex: 'statusName',
    width: '100px',
    align: 'center',
    scopedSlots: {
      customRender: 'statusName'
    }
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '200px',
    align: 'center',
    fixed: 'right',
    scopedSlots: {
      customRender: 'action'
    }
  }
]
export const reformHistoryColumns = [{
  title: '责任单位',
  dataIndex: 'dutyUnitName',
  align: 'center',
  width: '250px'
},
{
  title: '负责整改数',
  dataIndex: 'perilRectifySum',
  align: 'center',
  width: '300px'
},
{
  title: '已整改数',
  dataIndex: 'perilRectifiedNum',
  width: '150px',
  align: 'center'
},
{
  title: '未通过数',
  dataIndex: 'perilNotRectifiedNum',
  align: 'center',
  width: '200px'
},
{
  title: '整改率',
  dataIndex: 'rateOfPerilRectified',
  width: '200px',
  align: 'center',
  scopedSlots: {
    customRender: 'rateOfPerilRectified'
  }
},
{
  title: '未通过率',
  dataIndex: 'rateOfPerilNotRectified',
  align: 'center',
  width: '150px',
  scopedSlots: {
    customRender: 'rateOfPerilNotRectified'
  }
}
]
export const investColumns = [
  // {
  //   title: '填报单位',
  //   dataIndex: 'dutyUnitName',
  //   width: '200px',
  //   align: 'center'
  // },
  // {
  //   title: '填报人',
  //   dataIndex: 'informantName',
  //   width: '150px',
  //   align: 'center'
  // },
  {
    title: '隐患描述',
    dataIndex: 'perilDescription',
    width: '300px',
    align: 'center'
  },
  {
    title: '隐患等级',
    dataIndex: 'levelName',
    width: '150px',
    align: 'center'
  },
  {
    title: '隐患类型',
    dataIndex: 'typeName',
    width: '200px',
    align: 'center'
  },
  {
    title: '检查时间',
    dataIndex: 'checkTime',
    width: '200px',
    align: 'center'
  },
  // {
  //   title: '整改责任人',
  //   dataIndex: 'dutyPersonName',
  //   width: '150px',
  //   align: 'center'
  // },
  {
    title: '整改期限',
    dataIndex: 'abarbeitungDeadline',
    width: '200px',
    align: 'center'
  },
  {
    title: '复查人',
    dataIndex: 'reviewPersonName',
    width: '150px',
    align: 'center'
  },
  {
    title: '状态',
    dataIndex: 'statusName',
    align: 'center',
    width: '150px',
    scopedSlots: {
      customRender: 'statusName'
    }
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '120px',
    align: 'center',
    fixed: 'right',
    scopedSlots: {
      customRender: 'action'
    }
  }
]
// contractor
export const contractorColumns = [
  {
    title: '序号',
    scopedSlots: {
      customRender: 'serial'
    },
    dataIndex: 'index',
    width: '80px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '承包商名称',
    dataIndex: 'name',
    width: '200px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '性质',
    dataIndex: 'natureName',
    width: '150px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '地址',
    dataIndex: 'address',
    width: '250px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '是否在合作期内',
    dataIndex: 'isCooperationPeriod',
    width: '150px',
    align: 'center',
    customRender: (text) => {
      return text ? '不是' : '是'
    }
  },
  {
    title: '状态',
    dataIndex: 'status',
    width: '80px',
    align: 'center',
    scopedSlots: {
      customRender: 'status'
    }
    // customRender: (text) => {
    //   return text ? '已发布' : '未发布'
    // }
  },
  {
    title: '创建时间',
    dataIndex: 'createTime',
    width: '120px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '180px',
    align: 'center',
    // fixed: 'right',
    scopedSlots: {
      customRender: 'action'
    }
  }
]
// 承包商人员信息
export const contractorPersonColumns = [
  {
    title: '序号',
    scopedSlots: {
      customRender: 'serial'
    },
    dataIndex: 'index',
    width: '80px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '承包商名称',
    dataIndex: 'contractorName',
    width: '150px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '姓名',
    dataIndex: 'name',
    width: '150px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '身份证号',
    dataIndex: 'idCard',
    width: '250px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '联系方式',
    dataIndex: 'contactWay',
    width: '150px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '岗位角色',
    dataIndex: 'post',
    width: '150px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '资质证书',
    dataIndex: 'qualification',
    width: '100px',
    align: 'center',
    scopedSlots: {
      customRender: 'viewAction'
    }
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '180px',
    align: 'center',
    scopedSlots: {
      customRender: 'action'
    }
  }
]
// 承包商违规记录
export const contractorIllegalColumns = [
  {
    title: '序号',
    scopedSlots: {
      customRender: 'serial'
    },
    dataIndex: 'index',
    width: '80px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '承包商名称',
    dataIndex: 'contractorName',
    width: '160px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '违规人',
    dataIndex: 'illegalPersonName',
    width: '100px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '违规时间',
    dataIndex: 'illegalDate',
    width: '190px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '违规描述',
    dataIndex: 'description',
    align: 'center',
    width: '200px',
    ellipsis: true
  },
  {
    title: '违规类型',
    dataIndex: 'typeName',
    width: '150px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '现场照片',
    dataIndex: 'localePhotos',
    width: '100px',
    align: 'center',
    ellipsis: true,
    scopedSlots: {
      customRender: 'viewAction'
    }
  },
  {
    title: '创建人',
    dataIndex: 'gmtCreateName',
    width: '100px',
    ellipsis: true,
    align: 'center'
  },
  {
    title: '创建时间',
    dataIndex: 'createTime',
    width: '190px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '处理结果',
    dataIndex: 'resultTypeName',
    width: '100px',
    align: 'center',
    ellipsis: true,
    scopedSlots: {
      customRender: 'resultTypeName'
    }
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '180px',
    align: 'center',
    // fixed: 'right',
    scopedSlots: {
      customRender: 'action'
    }
  }
]
// culture
export const cultureColumns = [{
  title: '序号',
  customRender: (text, record, index) => `${index + 1}`,
  dataIndex: 'number',
  width: '80px',
  align: 'center'
},
{
  title: '公司名称',
  dataIndex: 'entName',
  align: 'center'
},
{
  title: '资料名称',
  dataIndex: 'name',
  align: 'center'
},
{
  title: '附件',
  dataIndex: 'docName',
  align: 'center',
  scopedSlots: {
    customRender: 'docName'
  }
},
{
  title: '操作',
  dataIndex: 'action',
  width: '120px',
  align: 'center',
  scopedSlots: {
    customRender: 'action'
  }
}
]

// ticket
export const ticketColumns = [
  // {
  //   title: '序号',
  //   dataIndex: 'number',
  //   align: 'center',
  //   width: '60px',
  //   scopedSlots: { customRender: 'serial' }
  // },
  {
    title: '作业类别',
    dataIndex: 'ticketTypeName',
    align: 'center',
    width: '120px',
    ellipsis: true
  },
  {
    title: '作业票编号',
    dataIndex: 'code',
    align: 'center',
    width: '200px',
    ellipsis: true
  },
  {
    title: '申请单位',
    dataIndex: 'applicantUnitName',
    align: 'center',
    width: '200px',
    ellipsis: true
  },
  {
    title: '申请人',
    dataIndex: 'applicantName',
    align: 'center',
    width: '150px',
    ellipsis: true
  },
  {
    title: '计划作业起止时间',
    dataIndex: 'planEndTime',
    align: 'center',
    width: '320px',
    scopedSlots: {
      customRender: 'planEndTime'
    },
    ellipsis: true
  },
  {
    title: '作业状态',
    dataIndex: 'status',
    align: 'center',
    scopedSlots: {
      customRender: 'taskStatus'
    },
    ellipsis: true,
    width: '100px'
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '220px',
    align: 'center',
    scopedSlots: {
      customRender: 'action'
    }
  }
]
export const ticketRecordColumns = [
  {
  title: '序号',
  dataIndex: 'number',
  scopedSlots: {
    customRender: 'serial'
  },
  width: '60px',
  align: 'center'
},
{
  title: '作业类别',
  dataIndex: 'ticketType',
  width: '150px',
  align: 'center'
},
{
  title: '申请单位',
  dataIndex: 'applyUnit',
  width: '200px',
  align: 'center'
},
{
  title: '申请人',
  dataIndex: 'applyPerson',
  width: '150px',
  align: 'center'
},
{
  title: '作业内容',
  dataIndex: 'taskContent',
  width: '150px',
  align: 'center'
},
{
  title: '作业时间',
  dataIndex: 'time',
  align: 'center',
  width: '350px',
  scopedSlots: {
    customRender: 'time'
  }
},
{
  title: '作业地点',
  dataIndex: 'taskArea',
  width: '150px',
  align: 'center'
},
{
  title: '流程状态',
  dataIndex: 'processStatus',
  align: 'center',
  width: '150px',
  scopedSlots: {
    customRender: 'processStatus'
  }
},
{
  title: '操作',
  dataIndex: 'action',
  width: '150px',
  align: 'center',
  fixed: 'right',
  scopedSlots: {
    customRender: 'action'
  }
}
]

// alteration
export const alterationColumns = [
  {
  title: '序号',
  customRender: (text, record, index) => `${index + 1}`,
  dataIndex: 'number',
  align: 'center',
  width: '70px'
},
{
  title: '变更名称',
  dataIndex: 'name',
  width: '150px',
  align: 'center'
},
{
  title: '变更内容',
  dataIndex: 'content',
  width: '200px',
  align: 'center'
},
// {
//   title: '变更原因和目的',
//   dataIndex: 'reason',
//   width: '150px'
// },
{
  title: '变更单位',
  dataIndex: 'changeUnitName',
  width: '150px',
  align: 'center'
},
{
  title: '变更区域',
  dataIndex: 'areaName',
  width: '150px',
  align: 'center'

},
{
  title: '变更级别',
  dataIndex: 'levelName',
  width: '100px',
  align: 'center'
},
{
  title: '变更类型',
  dataIndex: 'typeName',
  width: '100px',
  align: 'center'
},
// {
//   title: '变更文件',
//   dataIndex: 'fileName',
//   align: 'center',
//   width: '150px',
//   scopedSlots: { customRender: 'fileName' }
// },
// {
//   title: '创建时间',
//   dataIndex: 'createTime',
//   width: '200px'
// },
// {
//   title: '起始时间',
//   dataIndex: 'startTime',
//   width: '200px'
// },
{
  title: '截止时间',
  dataIndex: 'endTime',
  width: '200px',
  align: 'center'
},
// {
//   title: '联系人',
//   dataIndex: 'contact'
// },
// {
//   title: '联系人电话',
//   dataIndex: 'mobile'
// },
// {
//   title: '备注',
//   dataIndex: 'remark'
// },
{
  title: '操作',
  dataIndex: 'action',
  width: '180px',
  align: 'center',
  fixed: 'right',
  scopedSlots: {
    customRender: 'action'
  }
}
]
// accident
export const accidentColumns = [
  {
  title: '序号',
  customRender: (text, record, index) => `${index + 1}`,
  dataIndex: 'number',
  width: '60px',
  align: 'center'
},
{
  title: '事故名称',
  dataIndex: 'name',
  width: '150px',
  align: 'center'
},
// {
//   title: '事故描述',
//   dataIndex: 'description',
//   width: '150px',
//   align: 'center'
// },
{
  title: '事故区域',
  dataIndex: 'areaName',
  width: '150px',
  align: 'center'
},
{
  title: '发生地点',
  dataIndex: 'detailLocation',
  width: '150px',
  align: 'center'
},
{
  title: '事故责任单位',
  dataIndex: 'chargerUnitName',
  width: '200px',
  align: 'center'
},
{
  title: '事故类别',
  dataIndex: 'typeName',
  width: '150px',
  align: 'center'
},
{
  title: '事故级别',
  dataIndex: 'levelName',
  width: '150px',
  align: 'center'
},
{
  title: '事故文件',
  dataIndex: 'reportName',
  align: 'center',
  width: '150px',
  scopedSlots: {
    customRender: 'reportName'
  }
},
// {
//   title: '创建人',
//   dataIndex: 'userName',
//   width: '150px',
//   align: 'center'
// },
{
  title: '发生时间',
  dataIndex: 'occurTime',
  width: '200px',
  align: 'center'
},
// {
//   title: '创建时间',
//   dataIndex: 'createTime',
//   width: '200px',
//   align: 'center'
// },
// {
//   title: '联系人',
//   dataIndex: 'contact'
// },
// {
//   title: '联系人电话',
//   dataIndex: 'mobile'
// },
// {
//   title: '备注',
//   dataIndex: 'remark'
// },
{
  title: '操作',
  dataIndex: 'action',
  width: '180px',
  align: 'center',
  fixed: 'right',
  scopedSlots: {
    customRender: 'action'
  }
}
]

// Meeting

export const meetingColumns = [
  {
    title: '序号',
    scopedSlots: {
      customRender: 'serial'
    },
    dataIndex: 'index',
    width: '80px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '会议名称',
    dataIndex: 'meetingsName',
    width: '200px',
    align: 'center',
    ellipsis: true
  },
  // {
  //   title: '会议类型',
  //   dataIndex: 'meetingsTypeName',
  //   width: '150px',
  //   align: 'center',
  //   ellipsis: true
  // },
  {
    title: '会议时间',
    dataIndex: 'meetingsTime',
    width: '150px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '主持人',
    dataIndex: 'meetingsHosterName',
    width: '120px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '记录人',
    dataIndex: 'gmtCreateName',
    width: '120px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '180px',
    align: 'center',
    // fixed: 'right',
    scopedSlots: {
      customRender: 'action'
    }
  }
]

// Activity
export const activityColumns = [
  // {
  //   title: '序号',
  //   scopedSlots: {
  //     customRender: 'serial'
  //   },
  //   dataIndex: 'index',
  //   width: '80px',
  //   align: 'center',
  //   ellipsis: true
  // },
  {
    title: '活动名称',
    dataIndex: 'activityName',
    width: '200px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '活动类型',
    dataIndex: 'activityTypeName',
    width: '150px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '负责人',
    dataIndex: 'managerName',
    width: '120px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '组织单位',
    dataIndex: 'deptName',
    width: '200px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '活动时间',
    dataIndex: 'activityTime',
    width: '280px',
    align: 'center',
    customRender: (text, record) => {
      return record.planBgTime + '~' + record.planEndTime
    },
    ellipsis: true
  },
  {
    title: '状态',
    dataIndex: 'status',
    width: '80px',
    align: 'center',
    scopedSlots: {
      customRender: 'status'
    }
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '250px',
    align: 'center',
    scopedSlots: {
      customRender: 'action'
    }
  }
]

export const otherTaskColumns = [
  {
  title: '作业证编号',
  dataIndex: 'code',
  width: '200px',
  align: 'center'
},
// {
//   title: '申请单位',
//   dataIndex: 'applicantUnitName',
//   width: '150px',
//   align: 'center',
//   ellipsis: true
// },
{
  title: '作业类型',
  dataIndex: 'workTypeName',
  width: '150px',
  align: 'center'
},
// {
//   title: '作业地点',
//   dataIndex: 'workAddress',
//   align: 'center',
//   width: '200px',
//   ellipsis: true
// },
// {
//   title: '作业时间',
//   dataIndex: 'planStartTime',
//   align: 'center',
//   scopedSlots: {
//     customRender: 'time'
//   }
// }
]

export const simpleTaskColumns = [
  {
    title: '作业类型',
    dataIndex: 'workTypeName',
    width: '150px'
  },
  {
    title: '作业证编号',
    dataIndex: 'code',
    width: '200px'
  }
]

export const personTableColumns = [{
  title: '姓名',
  dataIndex: 'personName',
  align: 'center'
},
{
  title: '单位',
  dataIndex: 'deptName',
  align: 'center'
},
{
  title: '证件编号',
  dataIndex: 'certificateNumber',
  align: 'center'
},
{
  title: '证件有效期',
  dataIndex: 'validTime',
  align: 'center'
},
{
  title: '操作',
  dataIndex: 'operation',
  align: 'center',
  fixed: 'right',
  scopedSlots: {
    customRender: 'operation'
  }
}
]
export const taskTypes = [{
  title: '动火作业',
  name: 'HotWork',
  type: 0,
  src: require('@/assets/tickets/hot.png')
},
{
  title: '高处作业',
  name: 'HighWork',
  type: 1,
  src: require('@/assets/tickets/high.png')
},
{
  title: '盲板抽堵',
  name: 'BlindPlateWork',
  type: 6,
  src: require('@/assets/tickets/blind-plate.png')
},
{
  title: '受限空间',
  name: 'ConfinedSpaceWork',
  type: 3,
  src: require('@/assets/tickets/confined-space.png')
},
{
  title: '吊装作业',
  name: 'HoistingWork',
  type: 4,
  src: require('@/assets/tickets/lifting.png')
},
{
  title: '临时用电',
  name: 'ElectricWork',
  type: 2,
  src: require('@/assets/tickets/electric.png')
},
{
  title: '动土作业',
  name: 'BuildingWork',
  type: 5,
  src: require('@/assets/tickets/building.png')
},
{
  title: '断路作业',
  name: 'BreakingWork',
  type: 7,
  src: require('@/assets/tickets/road-breaking.png')
},
{
  title: '检修作业',
  name: 'RepairWork',
  type: 8,
  src: require('@/assets/tickets/repair.png')
}
]
