// spang-web/src/views/safety/task/hoisting/risks.js
export const hotWorkRiskFillinList = [
    {
      titleId: 1,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '作业前对作业人员内进行安全教育',
      measureLabel: '',
      signDate: ''
    },
    {
      titleId: 2,
      isTitle: 1,
      titleValue: '',
      isMeasures: 0,
      label: '吊装质量大于等于10吨的物体和土建工程主体结构；吊物虽不足10吨，但形状复杂， 刚度小，长径比大，精密贵重，施工条件特殊，需编制吊装作业方案，并经作业主管 部门和安全环保部审查，报主管副总经理或总工程师批准后方可实施。',
      measureLabel: '',
      measuresValue: '',
      measuresText: ''
    },
    {
      titleId: 3,
      isTitle: 1,
      titleValue: '',
      isMeasures: 0,
      measuresValue: '',
      measuresText: '',
      label: '指派专人监护，并坚守岗位，非作业人员禁止入内。',
      measureLabel: ''
    },
    {
      titleId: 4,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '作业人员已按规定佩戴防护器具和个体防护用品。',
      measureLabel: ''
    },
    {
      titleId: 5,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '应事先与车间负责人取得联系，建立联系信号。',
      measureLabel: ''
    },
    {
      titleId: 6,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '在吊装作业现场设置安全警戒标志，无关人员不许进入作业现场。',
      measureLabel: '',
    },
    {
      titleId: 7,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '夜间作业要有足够的照明。',
      measureLabel: ''
    },
    {
      titleId: 8,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '室外作业遇到大雪、暴雨、大雾及六级以上大风，停止作业。',
      measureLabel: ''
    },
    {
      titleId: 9,
      isTitle: 1,
      titleValue: '',
      isMeasures: 0,
      measuresText: '',
      label: '检查起重吊装设备、钢丝绳、揽风绳、链条、吊钩等各种机具，保证安全可靠。',
      measureLabel: ''
    },
    {
      titleId: 10,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '应分工明确，坚守岗位，并按规定的联络信号，统一指挥。',
      measureLabel: ''
    },
    {
      titleId: 11,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '将建筑物、构筑物作为锚点，需经技术部审查核算并批准。',
      measureLabel: ''
    },
    {
      titleId: 12,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '吊装绳索、揽风绳、拖拉绳等避免与带电线路接触，并保持安全距离。',
      measureLabel: ''
    },
    {
      titleId: 13,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '人员随同吊装重物或吊装机械升降，应采取可靠的安全措施，并经过现场指挥人员 批准。',
      measureLabel: ''
    },
    {
      titleId: 14,
      isTitle: 1,
      titleValue: '',
      isMeasures: 0,
      measuresText: '',
      label: '利用管道、管架、电杆、机电设备等做吊装锚点，不准吊装。',
      measureLabel: ''
    },
    {
      titleId: 15,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '悬空吊物下方站人、通行或工作，不准吊装。',
      measureLabel: ''
    },
    {
      titleId: 16,
      isTitle: 1,
      titleValue: '',
      isMeasures: 0,
      measuresText: '',
      label: '超负荷或吊物质量不明，不准吊装。',
      measureLabel: ''
    },
    {
      titleId: 17,
      isTitle: 1,
      titleValue: '',
      isMeasures: 0,
      measuresText: '',
      label: '斜拉重物、重物埋在地下或重物紧固不牢，绳打结、绳不齐，不准吊装。',
      measureLabel: ''
    },
    {
      titleId: 18,
      isTitle: 1,
      titleValue: '',
      isMeasures: 0,
      measuresText: '',
      label: '棱角物体没有衬垫措施，不准吊装。',
      measureLabel: ''
    },
    {
      titleId: 19,
      isTitle: 1,
      titleValue: '',
      isMeasures: 0,
      measuresText: '',
      label: '安全装置失灵，不准吊装。',
      measureLabel: ''
    },
    {
      titleId: 20,
      isTitle: 1,
      titleValue: '',
      isMeasures: 0,
      measuresText: '',
      label: '用定型起重吊装机械（履带吊、轮胎吊、轿式吊车等）进行吊装作业，遵守该定型机 械的操作规程。',
      measureLabel: ''
    },
    {
      titleId: 21,
      isTitle: 1,
      titleValue: '',
      isMeasures: 0,
      measuresText: '',
      label: '作业过程，应先用低高度，。短行程试吊。',
      measureLabel: ''
    },
    {
      titleId: 22,
      isTitle: 1,
      titleValue: '',
      isMeasures: 0,
      measuresText: '',
      label: '作业过程中出现危险品泄漏，立即停止作业，撤离人员。',
      measureLabel: ''
    },
    {
      titleId: 23,
      isTitle: 1,
      titleValue: '',
      isMeasures: 0,
      measuresText: '',
      label: '作业完成后现场清理现场杂物。',
      measureLabel: ''
    },
    {
      titleId: 24,
      isTitle: 1,
      titleValue: '',
      isMeasures: 0,
      measuresText: '',
      label: '作业人员持有法定的有效证件。',
      measureLabel: ''
    },
    {
      titleId: 25,
      isTitle: 1,
      titleValue: '',
      isMeasures: 0,
      measuresText: '',
      label: '地下通讯电（光）缆、局域网络电（光）缆、排水沟的盖板，承重吊装机械的负重已 确认，保护措施已落实。',
      measureLabel: ''
    },
    {
      titleId: 26,
      isTitle: 1,
      titleValue: '',
      isMeasures: 0,
      measuresText: '',
      label: '起吊物的质量已确认，在吊装机械的称重范围内。',
      measureLabel: ''
    },
    {
      titleId: 27,
      isTitle: 1,
      titleValue: '',
      isMeasures: 0,
      measuresText: '',
      label: '在吊装高度的管线、桥架已做好防护措施。',
      measureLabel: ''
    },
    {
      titleId: 28,
      isTitle: 1,
      titleValue: '',
      isMeasures: 0,
      measuresText: '',
      label: '作业现场围栏、警戒线、警告牌、夜间警示灯已按要求设置。',
      measureLabel: ''
    },
    {
      titleId: 29,
      isTitle: 1,
      titleValue: '',
      isMeasures: 0,
      measuresText: '',
      label: '作业高度和转臂范围内，无架空线路。',
      measureLabel: ''
    },
    {
      titleId: 30,
      isTitle: 1,
      titleValue: '',
      isMeasures: 0,
      measuresText: '',
      label: '人员出入口和撤离安全措施已落实：A：指示牌；B：指示灯。',
      measureLabel: ''
    },
    {
      titleId: 31,
      isTitle: 1,
      titleValue: '',
      isMeasures: 0,
      measuresText: '',
      label: '在爆炸危险生产区域内作业，机动车排气管已安装火星熄灭器。',
      measureLabel: ''
    },
    {
      titleId: 32,
      isTitle: 1,
      titleValue: '',
      isMeasures: 0,
      measuresText: '',
      label: '现场夜间有充足照明：A：36V，24V，12V防水型灯；B：36V，24V，12V防爆型灯。',
      measureLabel: ''
    },
    {
      titleId: 33,
      isTitle: 1,
      titleValue: '',
      isMeasures: 0,
      measuresText: '',
      label: '有危险有害因素辨识人，安全措施确认人。',
      measureLabel: ''
    },
    {
      titleId: 34,
      isTitle: 1,
      titleValue: '',
      isMeasures: 0,
      measuresText: '',
      label: '现场有专人监护。',
      measureLabel: ''
    },
    {
      titleId: 35,
      isTitle: 1,
      titleValue: '',
      isMeasures: 0,
      measuresText: '',
      label: '补充措施',
      measureLabel: 'false'
    }
  ]
