// spang-web/src/api/accident.js
const AccidentApi = {
  accidentExpress: '/spang-safety/accidents/accidentexpress',
  accidentExpressList: '/spang-safety/accidents/accidentexpress/getPageList',
  accidentExpressUpdate: '/spang-safety/accidents/accidentexpress/updateStatusById',
  accidentExpressDetail: '/spang-safety/accidents/accidentexpressdetails',
  accidentExpressDetailList: '/spang-safety/accidents/accidentexpressdetails/getList',
  accidentNearmiss: '/spang-safety/accidents/nearmissaccident',
  accidentnearmissList: '/spang-safety/accidents/nearmissaccident/getPageList',
  accidentNearmissUpdate: '/spang-safety/accidents/nearmissaccident/updateStatusById',
  accidentNearmissApprove: '/spang-safety/accidents/nearmissaccident/updateApproveByIds',
  accidentReport: '/spang-safety/accidents/accidentreport',
  accidentReportList: '/spang-safety/accidents/accidentreport/getPageList',
  accidentReportUpdate: '/spang-safety/accidents/accidentreport/updateStatusById',
  accidentReportApprove: '/spang-safety/accidents/accidentreport/updateApproveByIds',
  accidentProcessList: '/spang-safety/accidents/accidentreport/getProcessPageList',
  accidentProcess: '/spang-safety/accidents/accidentprocess',
  accidentArchiveList: '/spang-safety/accidents/accidentreport/getArchivePageList',
  accidentArchive: '/spang-safety/accidents/accidentreport/getArchiveById'
}

export default AccidentApi
