// spang-web/src/api/electric-work.js
import request from '@/utils/request'
const url = {
  work: '/spang-safety/tickets/electricityticket/main',
  history: '/spang-safety/tickets/electricityticket/peoplehis/loadPeopleHisListByMainId',
  approveHistory: '/spang-safety/tickets/electricityticket/approvehis/loadApproveHisList',
  safety: '/spang-safety/tickets/electricityticket/safefillin',
  risk: '/spang-safety/tickets/electricityticket/riskfillin'
}
// 提交数据 （基础表单）
export const saveForm = (params, method = 'post') => {
  return request({
    url: url.work,
    method,
    data: params
  })
}

// 根据id 查看详情 (基础信息)
export const getDetail = (id) => {
  return request({
    url: url.work + `?id=${id}`,
    method: 'get'
  })
}

// 人员更换历史  基础信息 && 作业单位
export const getPersonHistory = (params) => {
  const { mainId, current, size } = params
  return request({
    url: url.history + `?mainId=${mainId}&current=${current}&size=${size}`,
    method: 'get'
  })
}

// 审批历史
export const getApprovalHistory = (params) => {
  const { mainId, current, size } = params
  return request({
    url: url.approveHistory + `?mainId=${mainId}&current=${current}&size=${size}`,
    method: 'get'
  })
}
// 获取详情 (风险辨识)
export const getRiskIdentificationDetail = (params) => {
  return request({
    url: url.risk + '/getFullByMainId' + `?mainId=${params.id}`,
    method: 'get'
   })
 }

 // 获取详情 (安全交底)
 export const getSafetyFromDetail = (params) => {
   return request({
     url: url.safety + `?mainId=${params.id}`,
     method: 'get'
   })
 }
