// spang-web/src/api/high-work.js
import request from '@/utils/request'
const url = {
  highWork: '/spang-safety/safety/high-work-ticket'
}
export const saveForm = (data, method) => {
  return request({
    url: url.highWork,
    method: method,
		data
  })
}

export const getDetail = (id) => {
  return request({
    url: url.highWork + '/' + id,
    method: 'get'
  })
}

export const loadHistory = (id) => {
	return request({
	  url: url.highWork + '/user/history',
	  method: 'get',
    params: { id }
	})
}

export const getConfirmation = (data) => {
  return request({
    url: url.highWork + '/locale',
    method: 'get',
		params: data
  })
}

export const getApprovalHistory = (data) => {
	return request({
	  url: url.highWork + '/approvehis',
	  method: 'get',
		params: data
	})
}

export const assignWorker = (data) => {
	return request({
	  url: url.highWork + '/user',
	  method: 'post',
		data
	})
}

export const getAssignWorker = (data) => {
  return request({
	  url: url.highWork + '/user',
	  method: 'get',
		params: data
	})
}
