// spang-web/src/api/hoisting-work.js
import request from '@/utils/request'
const hotworkApi = {
  work: '/spang-safety/tickets/liftingwork/main',
  PersonHistory: '/spang-safety/tickets/liftingwork/peoplehis/loadPeopleHisListByMainId',
  HotworkPeopleWork: '/spang-safety/tickets/hotwork/peoplework',
  RiskIdentification: '/spang-safety/tickets/liftingwork/safefillin',
  SafetyForm: '/spang-safety/tickets/liftingwork/safefillin',
  GetAnalysisHistory: '/spang-safety/tickets/hotwork/analysis/loadAnalysisHisList',
  ApprovalHistory: '/spang-safety/tickets/liftingwork/approvehis/loadApproveHisList',
  GetInspectionHistory: '/spang-safety/tickets/hotwork/acceptcheck'
}

// 动火作业API Start
// 提交数据 （基础表单）
export const saveForm = (params, method = 'post') => {
  return request({
    url: hotworkApi.work,
    method,
    data: params
  })
}

// 根据id 查看详情 (基础信息)
export const getDetail = (id) => {
  return request({
    url: hotworkApi.work + `?id=${id}`,
    method: 'get'
  })
}
// 人员更换历史  基础信息 && 作业单位
export const getPersonHistory = (params) => {
  const { mainId, current, size } = params
  return request({
    url: hotworkApi.PersonHistory + `?mainId=${mainId}&current=${current}&size=${size}`,
    method: 'get'
  })
}

 // 新增&&修改 数据 （作业单位）
export const addHotworkPeople = (params) => {
  return request({
    url: hotworkApi.HotworkPeopleWork,
    method: 'put',
    data: params
  })
}

// 获取详情 （ 作业单位）
export const getHotworkPeopleDetail = (params) => {
  return request({
    url: hotworkApi.HotworkPeopleWork + `?mainId=${params.id}`,
    method: 'get'
  })
}
// 获取详情 (风险辨识)
export const getRiskIdentificationDetail = (params) => {
  return request({
    url: hotworkApi.RiskIdentification + '/getFullByMainId' + `?mainId=${params.id}`,
    method: 'get'
   })
 }

 // 获取详情 (安全交底)
 export const getSafetyFromDetail = (params) => {
   return request({
     url: hotworkApi.SafetyForm + `?mainId=${params.id}`,
     method: 'get'
   })
 }
 // 获取分析化验详情
 export const getAnalysisHistory = (params) => {
  return request({
    url: hotworkApi.GetAnalysisHistory + `?mainId=${params.mainId}&current=${params.current}&size=${params.size}`,
    method: 'get'
  })
}
// 审批历史
export const getApprovalHistory = (params) => {
  const { mainId, current, size } = params
  return request({
    url: hotworkApi.ApprovalHistory + `?mainId=${mainId}&current=${current}&size=${size}`,
    method: 'get'
  })
}
// 验收历史
export const getInspectionHistory = (params) => {
  return request({
    url: hotworkApi.GetInspectionHistory + `?mainId=${params.mainId}&current=${params.current}&size=${params.size}`,
    method: 'get'
  })
}
// 动火作业API End
