// spang-web/src/api/index.js
import { get } from '@/axios'
const BASE_URL = process.env.BASE_URL
// 获取组织机构根节点
export const GET_DEPT_ROOT = () => get(BASE_URL + 'api/spang-system/dept/lazy-tree?parentId=0')

// 根据部门id获取分页人员信息
export const GET_PAGE_EMPLOYEE = data => get(BASE_URL + 'api/spang-system/page', data)

// 获取组织机构子节点
export const GET_DEPT_TREE = data => get(BASE_URL + 'api/spang-system/dept/lazy-tree', data)

// 获取组织机构下人员信息
export const GET_USER_BY_DEPT = data => get(BASE_URL + 'userData.json', data)

export const SEARCH_USER = data => get('/api/spang-system/like', data)

export const GET_DATABASE = data => get('/api/spang-snaker/snaker-flow/datasource/list', data)

export const GET_DATA_TABLE = data => get('/api/spang-snaker/snaker-flow/datasource/tables', data)

export const GET_DATA_COLUMN = data => get('/api/spang-snaker/snaker-flow/datasource/columns', data)

export const GET_ROLE_TREE = data => get('/api/spang-system/role/lazy-tree', data)

export const GET_POSITION = data => get('/api/spang-system/post/select', data)
export const Start_Flow = data => get('/api/spang-snaker/snaker-flow/process/start', data)
