// spang-web/src/api/login.js
import request from '@/utils/request'
import website from '@/config/website'

export const userApi = {
  Login: '/spang-auth/oauth/token',
  Phone: '/auth/phone',
  Logout: '/spang-auth/oauth/logout',
  ForgePassword: '/auth/forge-password',
  Register: '/auth/register',
  SendSms: '/account/sms',
  SendImg: '/spang-auth/oauth/captcha',
  SendSmsErr: '/account/sms_err',
  // get my info
  // UserInfo: '/spang-auth/oauth/user-info',
  UserInfo: '/spang-system/info',
  // UserMenu: '/spang-system/menu/routes',
  UserMenu: '/spang-system/menu/biz/routes-children',
  SystemButtons: '/spang-system/menu/buttons'
}

export function loginByUsername (parameter) {
  const { tenantId, username, password, key, code } = parameter
  return request({
    url: userApi.Login,
    method: 'post',
    // data: parameter
    headers: {
      'Tenant-Id': tenantId,
      'Captcha-Key': key,
      'Captcha-Code': code
    },
    params: {
      tenantId,
      username,
      password,
      grant_type: (website.captchaMode ? 'captcha' : 'password'),
      scope: 'all'
    }
  })
}

export function getImgCaptcha () {
  return request({
    url: userApi.SendImg,
    method: 'get'
  })
}

export function loginByPhone (parameter) {
  return request({
    url: userApi.Phone,
    method: 'post',
    data: parameter
  })
}

export function getSmsCaptcha (parameter) {
  return request({
    url: userApi.SendSms,
    method: 'post',
    data: parameter
  })
}

export function getInfo () {
  return request({
    url: userApi.UserInfo,
    method: 'get'
  })
}

export function getCurrentUserNav () {
  return request({
    url: userApi.UserMenu,
    method: 'get'
  })
}

export function logout () {
  return request({
    url: userApi.Logout,
    method: 'get'
  })
}

export function getmenu (parameters) {
  return request({
    url: userApi.UserMenu,
    method: 'post',
    data: []
  })
}

export function getUserMenu (parameter) {
  const { id } = parameter
  return request({
    url: userApi.UserMenu,
    method: 'get',
    params: {
      id: id
    }
    // data: parameter
  })
}

export const getButtons = () => {
  return request({
    url: userApi.SystemButtons,
    method: 'get'
  })
}

export const refreshToken = (refresh_token, tenantId) => request({
  url: '/spang-auth/oauth/token',
  method: 'post',
  headers: {
    'Tenant-Id': tenantId
  },
  params: {
    tenantId,
    refresh_token,
    grant_type: "refresh_token",
    scope: "all",
  }
});
