// spang-web/src/api/system.js
import request from '@/utils/request'

const url = {
  dept: '/spang-system/dept/tree',
	dict: '/spang-system/dict-biz/dictionary',
	lazyDept: '/spang-system/dept/list',
	users: '/spang-system/user-list',
	info: '/spang-system/detail',
  UserAuthority: '/spang-system/getIfUserWithBizCode',
  usersWithPage: '/spang-system/getPageVOList'
}

export const getDepartments = () => {
  return request({
    url: url.dept,
    method: 'get'
  })
}

export const getDict = (code) => {
  return request({
    url: url.dict,
    method: 'get',
		params: {
			code
		}
  })
}

export const getLazyDept = (data) => {
  return request({
    url: url.lazyDept,
    method: 'get',
		data
  })
}

export const getUsers = (data) => {
  return request({
    url: url.users,
    method: 'get',
		params: data
  })
}

export const getUsersWithPage = (data) => {
  return request({
    url: url.usersWithPage,
    method: 'get',
		params: data
  })
}

export const getUserByDeptId = (id) => {
  return request({
    url: url.info,
    method: 'get',
		data: { id }
  })
}

export const getUserAuth = (params) => {
  const { bizCode, userId, deptId } = params
  return request({
    url: url.UserAuthority + `?bizCode=${bizCode}&userId=${userId}&deptId=${deptId}`,
    method: 'get'
  })
}
