// spang-web/src/api/urls.js
const apiURLs = {
  meetingList: '/spang-safety/safetyobcomm/safetymeetings/getPageList',
  meeting: '/spang-safety/safetyobcomm/safetymeetings',
  meetingStatusUpdate: '/spang-safety/safetyobcomm/safetymeetings/updateStatusById',

  // Activity
  activityList: '/spang-safety/safetyobcomm/safetyactivity/getPageList',
  activity: '/spang-safety/safetyobcomm/safetyactivity',
  activityStatusUpdate: '/spang-safety/safetyobcomm/safetyactivity/updateStatusById',

  // Hazop
  hazopList: '/spang-safety/risk/hazopreport/getPageList',
  hazop: '/spang-safety/risk/hazopreport',
  hazopStatusUpdate: '/spang-safety/risk/hazopreport/updateStatusById',

  // 区域管理
  areaTreeList: '/spang-safety/risk/areas/main/getTreeList',
  areaTree: '/spang-safety/risk/areas/main',
  areaPageList: '/spang-safety/risk/areas/details/getPageList',
  areaTable: '/spang-safety/risk/areas/details',

  // 风险管控层级
  levelList: '/spang-safety/risk/setup/controllevel/getList',
  level: '/spang-safety/risk/setup/controllevel',
  getRiskLevelByScore: '/spang-safety/risk/setup/controllevel/getResultVOByScores',

  // 风险评价方法设置
  evaluationList1: '/spang-safety/risk/setup/accidenthappen/getList',
  evaluationList2: '/spang-safety/risk/setup/accidentserious/getList',
  evaluationList3: '/spang-safety/risk/setup/accidentjudge/getList',
  evaluationList4: '/spang-safety/risk/setup/controllevel/getList',
  evaluation1: '/spang-safety/risk/setup/accidenthappen',
  evaluation2: '/spang-safety/risk/setup/accidentserious',
  evaluation3: '/spang-safety/risk/setup/accidentjudge',
  evaluation4: '/spang-safety/risk/setup/controllevel',

  // law
  lawList: '/spang-safety/securitysystem/lawsregulations/getPageList',
  law: 'spang-safety/securitysystem/lawsregulations',
  lawStatusUpdate: '/spang-safety/securitysystem/lawsregulations/updateStatusById',
  // rules
  rulesList: '/spang-safety/securitysystem/systemmgmt/getPageList',
  rules: 'spang-safety/securitysystem/systemmgmt',
  rulesStatusUpdate: '/spang-safety/securitysystem/systemmgmt/updateStatusById',
  // commitmentModelPerson
  commitmentModelPersonList: '/spang-safety/safetyresp/safetytemplateusers/getPageList',
  commitmentModelPerson: '/spang-safety/safetyresp/safetytemplateusers',
  // targetResponsibility
  targetResponsibilityList: '/spang-safety/safetyresp/safetytemplate/getPageList',
  targetResponsibility: '/spang-safety/safetyresp/safetytemplate',
  targetResponsibilityStatusUpdate: '/spang-safety/safetyresp/safetytemplate/updateStatusById',
  //maintenance
  maintenanceList: '/spang-safety/safetyresp/companytemplate/getPageList',
  maintenance: '/spang-safety/safetyresp/companytemplate',
  maintenanceStatusUpdate: '/spang-safety/safetyresp/companytemplate/updateStatusById',
  maintenanceCopyUpdate: '/spang-safety/safetyresp/companytemplate/updateCopyById',
  //3registry
  registryList: '/spang-safety/violation/page',
  registry: '/spang-safety/violation',
  //deal
  dealList: '/spang-safety/violation-deal/page',
  deal: '/spang-safety/violation-deal',
  //chemicals
  chemicalsList: '/spang-safety/chemical/page',
  chemicals: '/spang-safety/chemical',
  //costPlan
  costPlanList: '/spang-safety/security/cost/page',
  costPlan: '/spang-safety/security/cost',
  //costReceive
  costReceiveList: '/spang-safety/security/cost/receive/page',
  costReceive: '/spang-safety/security/cost/receive',
  //costStatistic
  costStatisticList: '/spang-safety/security/cost/count',
  //examination
  examinationList: '/spang-edu/train/plan/approve/list',
  //orientationplan
  orientationplanList:'/spang-edu/admin/train/plan/list',


  // 两重点一重大
  importantAll: '/spang-safety/risk/twoandone/getAll',
  important: '/spang-safety/risk/twoandone',
  importantTree: '/spang-safety/risk/twoandone/getTreeList',
  importantTable: '/spang-safety/risk/twoandone/getListByParentId',
  importantUpdateStatus: '/spang-safety/risk/twoandone/updateStatusById',

  // 企业信息
  companyInfo: '/spang-safety/risk/twoandone/companyinfo',

  contractorIllegalList: '/spang-safety/contractor/record/page',
  contractorIllegal: '/spang-safety/contractor/record',

  // 特种作业证书
  certificateAdd: '/spang-edu/admin/work/cert/insert',
  certificateUpdate: '/spang-edu/admin/work/cert/update',
  certificate: '/spang-edu/admin/work/cert/list',
  certificateById: '/spang-edu/admin/work/cert/selectById',
  certificateDelete: '/spang-edu/admin/work/cert/deleteById',
  certificateStatus: '/spang-edu/admin/work/cert/updateStatus',

  // 承包商
  illegalPersons: '/spang-safety/contractor/user/list',

  // 下载文件
  fileDownloadNew: '/spang-safety/fileserver/downLoadFileById',
  fileUploadNew: '/spang-safety/fileserver/fileUpload'
}
export default apiURLs
