// spang-web/src/views/safety/orientation/courseWareMgr/table.js
// 课件管理
export const CourseWareColumns = [{
    title: '课件名称',
    dataIndex: 'coursewareName',
    align: 'center',
    ellipsis: true,
    width: '250px'
  },
  {
    title: '课件类型',
    dataIndex: 'coursewareTypeName',
    width: '100px',
    align: 'center'
  },
  {
    title: '课程描述',
    dataIndex: 'coursewareDesc',
    align: 'center',
    ellipsis: true,
    width: '250px'
  },
  {
    title: '发布范围',
    dataIndex: 'publishRange',
    width: '100px',
    align: 'center',
    scopedSlots: {
      customRender: 'publishRange'
    }
  },
  {
    title: '状态',
    dataIndex: 'status',
    width: '120px',
    align: 'center',
    scopedSlots: {
      customRender: 'status'
    }

  },
  {
    title: '创建人',
    dataIndex: 'createName',
    width: '120px',
    align: 'center'
  },
  {
    title: '创建时间',
    dataIndex: 'createTime',
    align: 'center',
    width: '150px'
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '250px',
    align: 'center',
    // fixed: 'right',
    scopedSlots: {
      customRender: 'action'
    }
  }
]
// 线上考试列表
export const OnlineExamColumns = [{
    title: '序号',
    width: '150px',
    customRender: (text, record, index) => `${index + 1}`
  },
  {
    title: '考试名称',
    dataIndex: 'mainName',
    align: 'center'
  },
  {
    title: '考试类型',
    dataIndex: 'levelName',
    width: '120px',
    align: 'center'
  },
  {
    title: '考试时间',
    dataIndex: 'typeName',
    align: 'center',
    ellipsis: true
  },
  {
    title: '单位',
    dataIndex: 'dutyUserName',
    width: '120px',
    align: 'center'
  },
  {
    title: '创建人',
    dataIndex: 'dutyUserName',
    width: '120px',
    align: 'center'
  },
  {
    title: '创建时间',
    dataIndex: 'dutyUserName',
    align: 'center'
  },
  {
    title: '状态',
    dataIndex: 'dutyUserName',
    width: '120px',
    align: 'center'
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '200px',
    align: 'center',
    fixed: 'right',
    scopedSlots: {
      customRender: 'action'
    }
  }
]

export const onlineExamPersonColumns = [{
    title: '序号',
    width: '150px',
    customRender: (text, record, index) => `${index + 1}`
  },
  {
    title: '姓名',
    dataIndex: 'userName',
    align: 'center'
  },
  {
    title: '部门',
    dataIndex: 'deptName',
    align: 'center'
  },
  {
    title: '岗位',
    dataIndex: 'postName',
    align: 'center',
    ellipsis: true
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '200px',
    align: 'center',
    fixed: 'right',
    scopedSlots: {
      customRender: 'action'
    }
  }
]

// 试卷选择
export const PaperSelectColumns = [{
    title: '序号',
    width: '150px',
    customRender: (text, record, index) => `${index + 1}`
  },
  {
    title: '试卷名称',
    dataIndex: 'examName',
    align: 'center'
  },
  {
    title: '注意事项',
    dataIndex: 'remark',
    align: 'center',
    ellipsis: true
  }
]

// 培训计划选择
export const TrainSelectColumns = [{
    title: '序号',
    width: '150px',
    customRender: (text, record, index) => `${index + 1}`
  },
  {
    title: '名称',
    dataIndex: 'trainName',
    align: 'center',
    ellipsis: true
  },
  {
    title: '培训类型',
    dataIndex: 'trainType',
    align: 'center',
    ellipsis: true
  },
  {
    title: '培训级别',
    dataIndex: 'trainLevel',
    align: 'center',
    ellipsis: true
  },
  {
    title: '培训组织单位',
    dataIndex: 'orgUnitName',
    align: 'center',
    ellipsis: true
  },
  {
    title: '培训日期',
    align: 'center',
    customRender: (text, record) => {
      const times = [record.trainStartDate, record.trainEndDate]
      return times.filter(times => times).join('-')
    }
  }
]

// 查看线上考试人员列表
export const viewOnlineExamColumns = [{
    title: '序号',
    width: '150px',
    customRender: (text, record, index) => `${index + 1}`
  },
  {
    title: '姓名',
    dataIndex: 'userName',
    align: 'center',
    ellipsis: true
  },
  {
    title: '部门',
    dataIndex: 'deptName',
    align: 'center',
    ellipsis: true
  },
  {
    title: '岗位',
    dataIndex: 'postName',
    align: 'center',
    ellipsis: true
  },
  {
    title: '考试次数',
    dataIndex: 'supplementCount',
    align: 'center',
    ellipsis: true
  },
  {
    title: '成绩',
    dataIndex: 'maxScore',
    align: 'center',
    ellipsis: true
  },
  {
    title: '考试结果',
    dataIndex: 'examResult',
    align: 'center',
    ellipsis: true,
    scopedSlots: {
      customRender: 'examResult'
    }
  },
  {
    title: '提交时间',
    dataIndex: 'createTime',
    align: 'center',
    ellipsis: true
  }
]
