// spang-web/src/views/safety/task/hot/risks.js
export const hotWorkRiskFillinList = [
    {
      titleId: 1,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '可燃有毒介质名称$',
      measureLabel: '已吹扫置换清理干净',
      hotWorkRiskFillinList: [],
      signDate: ''
    },
    {
      titleId: 2,
      isTitle: 1,
      titleValue: '',
      isMeasures: 0,
      label: '在盛有或盛过化学品的容器、设备、管道等生产、储存装置上动火',
      measureLabel: '',
      measuresValue: '',
      showCheckbox: true,
      measuresText: '',
      options: [
        {
          label: '泄压',
          value: 0
        }, {
          label: '置换',
          value: 1
        }, {
          label: '蒸煮',
          value: 2
        }, {
          label: '水洗',
          value: 3
        }, {
          label: '吹扫',
          value: 4
        }, {
          label: '其他',
          value: 5
        }
      ],
      hotWorkRiskFillinList: []
    },
    {
      titleId: 3,
      isTitle: 1,
      titleValue: '',
      isMeasures: 0,
      measuresValue: '',
      measuresText: '',
      label: '动火设备、管道未与系统隔绝',
      measureLabel: '',
      showCheckbox: true,
      options: [
        {
          label: '加盲板$块',
          value: 0
        }, {
          label: '管线断开',
          value: 1
        }, {
          label: '法兰断开',
          value: 2
        }, {
          label: '其他',
          value: 3
        }
      ],
      hotWorkRiskFillinList: []
    },
    {
      titleId: 4,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '动火周围有易燃物，下水井、地漏、电缆沟等',
      measureLabel: '采取覆盖、铺沙、水封等手段进行隔离，并清除动火周围易燃物',
      hotWorkRiskFillinList: []
    },
    {
      titleId: 5,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '在通风不好或塔、油罐、容器内等不良环境下动火',
      measureLabel: '分析合格、测定含氧量、通风',
      hotWorkRiskFillinList: []
    },
    {
      titleId: 6,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '罐区内动火作业和倒罐、脱水作业同时进行',
      measureLabel: '同一罐区内不得同时进行倒罐或脱水作业',
      hotWorkRiskFillinList: []
    },
    {
      titleId: 7,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '作业火花飞溅',
      showMeasureLabel: true,
      measureLabel: '要采取围接、遮挡等措施，防止火花飞溅',
      hotWorkRiskFillinList: []
    },
    {
      titleId: 8,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '动火人、监火人及相关人员不清楚现场危险状况',
      measureLabel: '作业前已接受安全技术交底和安全教育',
      hotWorkRiskFillinList: []
    },
    {
      titleId: 9,
      isTitle: 1,
      titleValue: '',
      isMeasures: 0,
      measuresValue: [],
      measuresText: '',
      label: '现场未准备消防灭火器材',
      measureLabel: '',
      options: [
        {
          label: '配备消防蒸汽管$根',
          value: 0,
          textValues: []
        }, {
          label: '铁锹$把',
          value: 1,
          textValues: []
        }, {
          label: '灭火器$具',
          value: 2,
          textValues: []
        }, {
          label: '防火毡$块',
          value: 3,
          textValues: []
        }, {
          label: '消防水带$条',
          value: 4,
          textValues: []
        }
      ],
      hotWorkRiskFillinList: []
    },
    {
      titleId: 10,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '动火现场氧气瓶与乙炔气瓶之间、气瓶距动火点距离不够、烈日下暴晒；乙炔气瓶未直立放置',
      measureLabel: '氧气瓶与乙炔气瓶间距大于5m，距动火点距离大于10m，乙炔气瓶直立放置，并采取防暴晒措施',
      hotWorkRiskFillinList: []
    },
    {
      titleId: 11,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '电焊回路接线不正确',
      measureLabel: '回路线接在焊件上，把线不得穿过下水井或与其他设备搭接',
      hotWorkRiskFillinList: []
    },
    {
      titleId: 12,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '未对动火工器具进行检查',
      measureLabel: '未对动火工器具进行检查',
      hotWorkRiskFillinList: []
    },
    {
      titleId: 13,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '动火人未正确佩戴相应安全防护用品',
      measureLabel: '动火工器具检查合格',
      hotWorkRiskFillinList: []
    },
    {
      titleId: 14,
      isTitle: 1,
      titleValue: '',
      isMeasures: 0,
      measuresText: '',
      measuresValue: [],
      label: '5级以上大风室外作业',
      measureLabel: '禁止或停止动火作业',
      showCheckbox: true,
      options: [
        {
          label: '升级管理',
          value: 0
        }, {
          label: '清除易燃物',
          value: 1
        }, {
          label: '火花围接',
          value: 2
        }, {
          label: '遮挡',
          value: 3
        }, {
          label: '落实防坠落措施',
          value: 4
        }
      ],
      hotWorkRiskFillinList: []
    },
    {
      titleId: 15,
      isTitle: 1,
      titleValue: '',
      isMeasures: 1,
      label: '雨雪天气、地面有积水',
      measureLabel: '雨雪天气减少室外动火作业，必须动火时，清扫作业区积水，做好遮挡措施；焊机、电线等用电设备不得浸水',
      hotWorkRiskFillinList: []
    },
    {
      titleId: 16,
      isTitle: 1,
      titleValue: '',
      isMeasures: 0,
      measuresText: '',
      measuresValue: [],
      label: '作业人使用的防护用品',
      measureLabel: '禁止或停止动火作业',
      showCheckbox: true,
      options: [{
          label: '护目镜',
          value: 0
        }, {
          label: '焊帽',
          value: 1
        }, {
          label: '绝缘鞋',
          value: 2
        }, {
          label: '焊工手套',
          value: 3
        }, {
          label: '3M口罩',
          value: 4
        }, {
          label: '防毒面具',
          value: 5
        }, {
          label: '防毒衣',
          value: 6
        }, {
          label: '耳塞',
          value: 7
        }, {
          label: '安全带',
          value: 8
        }, {
          label: '防化服',
          value: 9
        }, {
          label: '隔热服',
          value: 10
        }, {
          label: '其它',
          value: 12
        }, {
          label: '长管空气呼吸器',
          value: 13
        }, {
          label: '正压式空气呼吸器',
          value: 14
        }],
      hotWorkRiskFillinList: []
    },
    {
      titleId: 17,
      isTitle: 1,
      titleValue: '',
      isMeasures: 0,
      measuresText: '',
      label: '补充措施',
      measureLabel: 'false',
      hotWorkRiskFillinList: []
    }
  ]
