// spang-web/src/views/safety/orientation/columns.js
// 线上考试列表
import Vue from 'vue'
import moment from 'moment'
const h = new Vue().$createElement
export const OnlineExamColumns = [
  {
    title: '考试名称',
    dataIndex: 'examName',
    align: 'center',
    width: '250px',
    ellipsis: true
  },
  {
    title: '考试类型',
    dataIndex: 'examType',
    width: '120px',
    align: 'center',
    ellipsis: true,
    scopedSlots: { customRender: 'examType' }
  },
  {
    title: '考试时间',
    dataIndex: 'typeName',
    align: 'center',
    width: '300px',
    ellipsis: true,
    customRender: (text, record) => {
      const time = [record.examStartTime, record.examEndTime].filter(item => !!(item)).join('至')
      return time || '-'
    }
  },
  {
    title: '单位',
    dataIndex: 'deptName',
    align: 'center',
    width: '120px',
    ellipsis: true
  },
  {
    title: '创建人',
    dataIndex: 'createName',
    width: '120px',
    ellipsis: true,
    align: 'center'
  },
  {
    title: '创建时间',
    dataIndex: 'createTime',
    width: '150px',
    ellipsis: true,
    align: 'center',
    customRender: (text, record) => {
      return moment(record.createTime).format('YYYY-MM-DD')
    }
  },
  {
    title: '状态',
    dataIndex: 'status',
    width: '100px',
    align: 'center',
    customRender: (text, record) => {
      if (!record.status) {
        return <a-tag color="green">已发布</a-tag>
      }
      return <a-tag color="orange">未发布</a-tag>
    }
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '200px',
    align: 'center',
    // fixed: 'right',
    scopedSlots: { customRender: 'action' }
  }
]

export const onlineExamPersonColumns = [
  {
    title: '序号',
    width: '150px',
    scopedSlots: { customRender: 'series' }
  },
  {
    title: '姓名',
    dataIndex: 'userName',
    align: 'center'
  },
  {
    title: '部门',
    dataIndex: 'deptName',
    align: 'center'
  },
  {
    title: '岗位',
    dataIndex: 'postName',
    align: 'center',
    ellipsis: true
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '200px',
    align: 'center',
    fixed: 'right',
    scopedSlots: { customRender: 'action' }
  }
]

// 试卷选择
export const PaperSelectColumns = [
  {
    title: '试卷名称',
    dataIndex: 'paperName',
    align: 'center',
    ellipsis: true
  },
  {
    title: '注意事项',
    dataIndex: 'remark',
    align: 'center',
    ellipsis: true
  }
]

// 培训计划选择
export const TrainSelectColumns = [
  {
    title: '名称',
    dataIndex: 'trainName',
    align: 'center',
    width: '150px',
    ellipsis: true
  },
  {
    title: '培训类型',
    dataIndex: 'trainTypeName',
    align: 'center',
    ellipsis: true
  },
  {
    title: '培训级别',
    dataIndex: 'trainLevelName',
    align: 'center',
    ellipsis: true
  },
  {
    title: '培训组织单位',
    dataIndex: 'orgUnitName',
    align: 'center',
    ellipsis: true
  },
  {
    title: '培训日期',
    align: 'center',
    width: '300px',
    ellipsis: true,
    customRender: (text, record) => {
      const times = [record.trainStartDate, record.trainEndDate]
      return times.filter(times => times).join('-')
    }
  }
]

// 查看线上考试人员列表
export const viewOnlineExamColumns = [
  {
    title: '序号',
    width: '80px',
    scopedSlots: { customRender: 'series' }
  },
  {
    title: '姓名',
    dataIndex: 'userName',
    align: 'center',
    width: '130px',
    ellipsis: true
  },
  {
    title: '部门',
    dataIndex: 'deptName',
    align: 'center',
    width: '130px',
    ellipsis: true
  },
  {
    title: '岗位',
    dataIndex: 'postName',
    align: 'center',
    width: '150px',
    ellipsis: true
  },
  {
    title: '考试次数',
    dataIndex: 'examCount',
    align: 'center',
    width: '100px',
    ellipsis: true
  },
  {
    title: '成绩',
    dataIndex: 'examScore',
    align: 'center',
    width: '100px',
    ellipsis: true
  },
  {
    title: '考试结果',
    dataIndex: 'examResult',
    align: 'center',
    width: '100px',
    ellipsis: true,
    scopedSlots: { customRender: 'examResult' }
  },
  {
    title: '提交时间',
    dataIndex: 'updateTime',
    align: 'center',
    ellipsis: true,
    width: '200px'
  }
]
// 线下考试列表
export const OfflineExamColumns = [
  {
    title: '考试名称',
    dataIndex: 'examName',
    align: 'center',
    width: '250px',
    ellipsis: true
  },
  {
    title: '培训计划',
    dataIndex: 'trainName',
    width: '120px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '考试日期',
    dataIndex: 'examStartTime',
    align: 'center',
    width: '200px',
    customRender: (text, record) => {
      if (record.examStartTime) {
        return record.examStartTime.split(' ')[0]
      }
      return ''
    }
  },
  {
    title: '单位',
    dataIndex: 'deptName',
    align: 'center',
    width: '120px',
    ellipsis: true
  },
  {
    title: '创建人',
    dataIndex: 'createName',
    width: '120px',
    ellipsis: true,
    align: 'center'
  },
  {
    title: '状态',
    dataIndex: 'status',
    width: '100px',
    align: 'center',
    customRender: (text, record) => {
      if (!record.status) {
        return <a-tag color="green">已发布</a-tag>
      }
      return <a-tag color="orange">未发布</a-tag>
    }
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '200px',
    align: 'center',
    // fixed: 'right',
    scopedSlots: { customRender: 'action' }
  }
]
// 个人考试计划查询列表
export const PersonalExamPlan = [
  {
    title: '考试名称',
    dataIndex: 'examName',
    ellipsis: true,
    align: 'center',
    width: '150px'
  },
  {
    title: '考试类型',
    dataIndex: 'examType',
    ellipsis: true,
    align: 'center',
    width: '150px',
    scopedSlots: { customRender: 'examType' }
  },
  {
    title: '考试时间',
    align: 'center',
    width: '300px',
    ellipsis: true,
    customRender: (text, record) => {
      const times = [record.examStartDate, record.examEndDate]
      return times.filter(times => times).join('-')
    }
  },
  {
    title: '考试成绩',
    dataIndex: 'score',
    ellipsis: true,
    align: 'center',
    width: '100px',
    customRender: (text, record) => {
      if (record.examResult != 0) {
        return text
      }
      return ''
    }
  },
  {
    title: '考试结果',
    dataIndex: 'examResult',
    ellipsis: true,
    align: 'center',
    width: '100px',
    scopedSlots: { customRender: 'examResult' }
  },
  {
    title: '查看',
    dataIndex: 'view',
    align: 'center',
    width: '100px',
    scopedSlots: { customRender: 'viewRecord' }
  },
  {
    title: '状态',
    dataIndex: 'examStatus',
    ellipsis: true,
    align: 'center',
    width: '100px',
    customRender: (text, record) => {
      if (record.examStatus === '0') {
        return <a-tag color="orange">未开始</a-tag>
      } else if (record.examStatus === '1') {
        return <a-tag color="green">已开始</a-tag>
      } else if (record.examStatus === '2') {
        return <a-tag color="blue">已考试</a-tag>
      } else if (record.examStatus === '3') {
        return <a-tag color="red">已过期</a-tag>
      }
      return ''
    }
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '200px',
    align: 'center',
    // fixed: 'right',
    scopedSlots: { customRender: 'action' }
  }
]
export const previewListColumns = [
   {
    title: '考试次数',
    dataIndex: 'examCount',
    align: 'center',
    width: '80px'
  },
  {
    title: '提交时间',
    dataIndex: 'createDate',
    ellipsis: true,
    align: 'center',
    width: '200px'
  },
   {
    title: '考试成绩',
    dataIndex: 'score',
    align: 'center',
    width: '100px'
  },
  {
    title: '考试结果',
    dataIndex: 'examResult',
    ellipsis: true,
    align: 'center',
    width: '100px',
    customRender: (text, record) => {
      if (record.examResult === 1) {
        return <a-tag color="red">不及格</a-tag>
      } else if (record.examResult === 2) {
        return '及格'
      } else {
        return ''
      }
    }
  },
   {
    title: '查看',
    ellipsis: true,
    align: 'center',
    width: '100px',
    scopedSlots: { customRender: 'viewRecord' }
  }
]
export const offlineExamPersonColumns = [
  {
    title: '序号',
    width: '100px',
    ellipsis: true,
    scopedSlots: { customRender: 'series' }
  },
  {
    title: '姓名',
    dataIndex: 'userName',
    align: 'center'
  },
  {
    title: '部门',
    dataIndex: 'deptName',
    align: 'center'
  },
  {
    title: '岗位',
    dataIndex: 'postName',
    align: 'center',
    ellipsis: true
  },
  {
    title: '成绩',
    dataIndex: 'examScore',
    align: 'center',
    ellipsis: true
  },
  {
    title: '考试结果',
    dataIndex: 'examResult',
    align: 'center',
    ellipsis: true,
    scopedSlots: { customRender: 'examResult' }
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '200px',
    align: 'center',
    fixed: 'right',
    scopedSlots: { customRender: 'action' }
  }
]

export const viewOfflineExamColumns = [
  {
    title: '序号',
    width: '100px',
    ellipsis: true,
    scopedSlots: { customRender: 'series' }
  },
  {
    title: '姓名',
    dataIndex: 'userName',
    align: 'center'
  },
  {
    title: '部门',
    dataIndex: 'deptName',
    align: 'center'
  },
  {
    title: '岗位',
    dataIndex: 'postName',
    align: 'center',
    ellipsis: true
  },
  {
    title: '成绩',
    dataIndex: 'examScore',
    align: 'center',
    ellipsis: true
  },
  {
    title: '考试结果',
    dataIndex: 'examResult',
    align: 'center',
    ellipsis: true,
    scopedSlots: { customRender: 'examResult' }
  }
]
