// spang-web/src/store/modules/system.js
import { setStore, getStore } from '@/utils/store'
import { getButtons } from '@/api/login'

const system = {
  state: {
    title: '',
    menuId: '/',
    menu: getStore({ name: 'menu' }) || [],
    permission: getStore({ name: 'permission' }) || {}
  },
  mutations: {
    SET_MENU: (state, menu) => {
      state.menu = menu
      setStore({ name: 'menu', content: state.menu })
    },
    SET_MENUID: (state, key) => {
      state.menuId = key
      setStore({ name: 'menuId', content: state.menuId })
    },
    SET_TITLE: (state, title) => {
      state.title = title
    },
    SET_PERMISSION: (state, permission) => {
      const result = []
      function getCode (list) {
        list.forEach(item => {
          if (typeof (item) === 'object') {
            const chiildren = item.children
            const code = item.code
            if (chiildren) {
              getCode(chiildren)
            } else {
              result.push(code)
            }
          }
        })
      }
      getCode(permission)
      state.permission = {}
      result.forEach(item => {
        state.permission[item] = true
      })
      setStore({ name: 'permission', content: state.permission })
    }
  },
  actions: {
    setTitle: ({ commit }, title) => {
      commit('SET_TITLE', title)
    },
    setMenuId: ({ commit }, key) => {
      commit('SET_MENUID', key)
    },
    setMenu: ({ commit, dispatch }, data) => {
      const { routers } = data
      // const routers = this.routers
      // console.log(this.$store.state.routers)
      const routes = routers.find(item => item.path)
      // const routes = routers
      const menus = (routes && routes.children) || []
      commit('SET_MENU', '')
      commit('SET_MENU', menus)
      dispatch('GetButtons')
    },
    GetButtons: ({ commit }) => {
      return new Promise(resolve => {
        // 获取buttons权限接口
        getButtons().then(res => {
          commit('SET_PERMISSION', res.data)
        })
        // const permissionData = [
        //   {
        //     name: '安全文化管理',
        //     code: 'culture',
        //     path: '/safety/culture',
        //     children: [
        //       {
        //         name: '新增',
        //         code: 'culture_add'
        //       },
        //       {
        //         name: '查看',
        //         code: 'culture_view'
        //       },
        //       {
        //         name: '删除',
        //         code: 'culture_delete'
        //       },
        //       {
        //         name: '批量删除',
        //         code: 'culture_deletes'
        //       },
        //       {
        //         name: '编辑',
        //         code: 'culture_edit'
        //       }
        //     ]
        //   },
        //   {
        //     name: '事故管理',
        //     code: 'accident',
        //     path: '/safety/accident',
        //     children: [
        //       {
        //         name: '新增',
        //         code: 'accident_add'
        //       },
        //       {
        //         name: '查看',
        //         code: 'accident_view'
        //       },
        //       {
        //         name: '删除',
        //         code: 'accident_delete'
        //       },
        //       {
        //         name: '批量删除',
        //         code: 'accident_deletes'
        //       },
        //       {
        //         name: '编辑',
        //         code: 'accident_edit'
        //       }
        //     ]
        //   },
        //   {
        //     name: '承包商管理',
        //     code: 'contractor',
        //     path: '/safety/contractor',
        //     children: [
        //       {
        //         name: '新增',
        //         code: 'contractor_add'
        //       },
        //       {
        //         name: '查看',
        //         code: 'contractor_view'
        //       },
        //       {
        //         name: '删除',
        //         code: 'contractor_delete'
        //       },
        //       {
        //         name: '批量删除',
        //         code: 'contractor_deletes'
        //       },
        //       {
        //         name: '编辑',
        //         code: 'contractor_edit'
        //       }
        //     ]
        //   },
        //   {
        //     name: '变更管理',
        //     code: 'alteration',
        //     path: '/safety/alteration',
        //     children: [
        //       {
        //         name: '新增',
        //         code: 'alteration_add'
        //       },
        //       {
        //         name: '查看',
        //         code: 'alteration_view'
        //       },
        //       {
        //         name: '删除',
        //         code: 'alteration_delete'
        //       },
        //       {
        //         name: '批量删除',
        //         code: 'alteration_deletes'
        //       },
        //       {
        //         name: '编辑',
        //         code: 'alteration_edit'
        //       }
        //     ]
        //   },
        //   {
        //     name: '风险管理',
        //     code: 'risk',
        //     children: [
        //       {
        //         name: 'Hazop评估报告台账',
        //         code: 'hazop',
        //         path: '/safety/risk/hazop',
        //         children: [
        //           {
        //             name: '新增',
        //             code: 'risk_hazop_add'
        //           },
        //           {
        //             name: '查看',
        //             code: 'risk_hazop_view'
        //           },
        //           {
        //             name: '删除',
        //             code: 'risk_hazop_delete'
        //           },
        //           {
        //             name: '批量删除',
        //             code: 'risk_hazop_deletes'
        //           },
        //           {
        //             name: '编辑',
        //             code: 'risk_hazop_edit'
        //           }
        //         ]
        //       },
        //       {
        //         name: '作业风险管理',
        //         code: 'risk_work',
        //         path: '/safety/risk/work',
        //         children: [
        //           {
        //             name: '新增',
        //             code: 'risk_work_add'
        //           },
        //           {
        //             name: '查看',
        //             code: 'risk_work_view'
        //           },
        //           {
        //             name: '删除',
        //             code: 'risk_work_delete'
        //           },
        //           {
        //             name: '批量删除',
        //             code: 'risk_work_deletes'
        //           },
        //           {
        //             name: '编辑',
        //             code: 'risk_work_edit'
        //           }
        //         ]
        //       },
        //       {
        //         name: '设备设施风险管理',
        //         code: 'risk_equipment',
        //         path: '/safety/risk/equipment',
        //         children: [
        //           {
        //             name: '新增',
        //             code: 'risk_equipment_add'
        //           },
        //           {
        //             name: '查看',
        //             code: 'risk_equipment_view'
        //           },
        //           {
        //             name: '删除',
        //             code: 'risk_equipment_delete'
        //           },
        //           {
        //             name: '批量删除',
        //             code: 'risk_equipment_deletes'
        //           },
        //           {
        //             name: '编辑',
        //             code: 'risk_equipment_edit'
        //           }
        //         ]
        //       },
        //       {
        //         name: '工艺风险管理',
        //         code: 'risk_technology',
        //         path: '/safety/risk/technology',
        //         children: [
        //           {
        //             name: '新增',
        //             code: 'risk_technology_add'
        //           },
        //           {
        //             name: '查看',
        //             code: 'risk_technology_view'
        //           },
        //           {
        //             name: '删除',
        //             code: 'risk_technology_delete'
        //           },
        //           {
        //             name: '批量删除',
        //             code: 'risk_technology_deletes'
        //           },
        //           {
        //             name: '编辑',
        //             code: 'risk_technology_edit'
        //           }
        //         ]
        //       }
        //     ]
        //   },
        //   {
        //     name: '安全制度管理',
        //     code: 'rules',
        //     path: '/safety/rules',
        //     children: [
        //       {
        //         name: '新增',
        //         code: 'rules_add'
        //       },
        //       {
        //         name: '查看',
        //         code: 'rules_view'
        //       },
        //       {
        //         name: '删除',
        //         code: 'rules_delete'
        //       },
        //       {
        //         name: '批量删除',
        //         code: 'rules_deletes'
        //       },
        //       {
        //         name: '编辑',
        //         code: 'rules_edit'
        //       }
        //     ]
        //   },
        //   {
        //     name: '法律法规',
        //     code: 'law',
        //     path: '/safety/law',
        //     children: [
        //       {
        //         name: '新增',
        //         code: 'law_add'
        //       },
        //       {
        //         name: '查看',
        //         code: 'law_view'
        //       },
        //       {
        //         name: '删除',
        //         code: 'law_delete'
        //       },
        //       {
        //         name: '批量删除',
        //         code: 'law_deletes'
        //       },
        //       {
        //         name: '编辑',
        //         code: 'law_edit'
        //       }
        //     ]
        //   },
        //   {
        //     name: '安全教育培训',
        //     code: 'orientation',
        //     children: [
        //       {
        //         name: '特殊工种证书',
        //         code: 'certificate',
        //         path: '/safety/orientation/certificate',
        //         children: [
        //           {
        //             name: '新增',
        //             code: 'certificate_add'
        //           },
        //           {
        //             name: '查看',
        //             code: 'certificate_view'
        //           },
        //           {
        //             name: '删除',
        //             code: 'certificate_delete'
        //           },
        //           {
        //             name: '批量删除',
        //             code: 'certificate_deletes'
        //           },
        //           {
        //             name: '编辑',
        //             code: 'certificate_edit'
        //           }
        //         ]
        //       },
        //       {
        //         name: '三级安全教育',
        //         code: 'safetyeducation',
        //         path: '/safety/orientation/safetyeducation',
        //         children: [
        //           {
        //             name: '新增',
        //             code: 'safetyeducation_add'
        //           },
        //           {
        //             name: '查看',
        //             code: 'safetyeducation_view'
        //           },
        //           {
        //             name: '删除',
        //             code: 'safetyeducation_delete'
        //           },
        //           {
        //             name: '批量删除',
        //             code: 'safetyeducation_deletes'
        //           },
        //           {
        //             name: '编辑',
        //             code: 'safetyeducation_edit'
        //           }
        //         ]
        //       },
        //       {
        //         name: '特殊工种证书',
        //         code: 'certificate',
        //         path: '/safety/orientation/certificate',
        //         children: [
        //           {
        //             name: '新增',
        //             code: 'certificate_add'
        //           },
        //           {
        //             name: '查看',
        //             code: 'certificate_view'
        //           },
        //           {
        //             name: '删除',
        //             code: 'certificate_delete'
        //           },
        //           {
        //             name: '批量删除',
        //             code: 'certificate_deletes'
        //           },
        //           {
        //             name: '编辑',
        //             code: 'certificate_edit'
        //           }
        //         ]
        //       }
        //     ]
        //   }
        // ]
        // commit('SET_PERMISSION', permissionData)
        resolve()
      })
    }
  }
}

export default system
