// spang-web/src/views/emergency/resources/table.js
export const suppliesColumns = [
  {
    title: '序号',
    dataIndex: 'id',
    key: 'id'
  },
  {
    title: '企业名称',
    dataIndex: 'company'
  },
  {
    title: '名称',
    dataIndex: 'name'
  },
  {
    title: '型号',
    dataIndex: 'model'
  },
  {
    title: '类型',
    dataIndex: 'type'
  },
  {
    title: '库存数量',
    dataIndex: 'count'
  },
  {
    title: '经度',
    dataIndex: 'longitude'
  },
  {
    title: '维度',
    dataIndex: 'latitude'
  },
  {
    title: '存放地点',
    dataIndex: 'location'
  },
  {
    title: '用途',
    dataIndex: 'use'
  },
  {
    title: '使用期限',
    dataIndex: 'useLimit'
  },
  {
    title: '联系人',
    dataIndex: 'contact'
  },
  {
    title: '联系人电话',
    dataIndex: 'mobile'
  },
  {
    title: '备注',
    dataIndex: 'remark'
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '150px',
    align: 'center',
    scopedSlots: { customRender: 'action' }
  }
]

export const expertsColumns = [
  {
    title: '机构名称',
    dataIndex: 'company'
  },
  {
    title: '专家姓名',
    dataIndex: 'name'
  },
  {
    title: '性别',
    dataIndex: 'sex'
  },
  {
    title: '出生日期',
    dataIndex: 'birthday'
  },
  {
    title: '学历',
    dataIndex: 'degree'
  },
  {
    title: '手机号码',
    dataIndex: 'mobile'
  },
  {
    title: '职称',
    dataIndex: 'title'
  },
  {
    title: '从事行业',
    dataIndex: 'profession'
  },
  {
    title: '专长领域',
    dataIndex: 'expertise'
  },
  {
    title: '工作单位',
    dataIndex: 'workplace'
  },
  {
    title: '工作岗位',
    dataIndex: 'jobPosition'
  },
  {
    title: '毕业学校',
    dataIndex: 'university'
  },
  {
    title: '联系电话',
    dataIndex: 'mobile'
  },
  {
    title: '电子邮箱',
    dataIndex: 'email'
  },
  {
    title: '备注',
    dataIndex: 'remark'
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '150px',
    align: 'center',
    scopedSlots: { customRender: 'action' }
  }
]
