// spang-web/src/views/safety/health/columns.js
export const healthPersonalProfileColumns = [
  {
    title: '序号',
    dataIndex: 'number',
    width: '60px',
    align: 'center',
    scopedSlots: { customRender: 'number' }
  },
  {
    title: '姓名',
    dataIndex: 'name',
    width: '200px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '档案编号',
    dataIndex: 'fileNumber',
    width: '180px',
    align: 'center'
  },
  {
    title: '性别',
    dataIndex: 'sex',
    width: '80px',
    align: 'center'
  },
  {
    title: '籍贯',
    dataIndex: 'nativePlace',
    width: '180px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '在职状态',
    dataIndex: 'status',
    width: '120px',
    align: 'center'
  },
  {
    title: '入职时间',
    dataIndex: 'entryWorkTime',
    width: '180px',
    align: 'center'
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '180px',
    align: 'center',
    // fixed: 'right',
    scopedSlots: { customRender: 'action' }
  }
]

export const occupationHistoryColumns = [
  {
    title: '起始时间',
    dataIndex: 'startTime',
    width: '200px',
    align: 'center',
    ellipsis: true,
    customRender: (text, record, index) => `${record.startTime.substr(0,10)}~${record.endTime.substr(0,10)}`
  },
  {
    title: '工作单位',
    dataIndex: 'workUnit',
    width: '180px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '工种',
    dataIndex: 'workType',
    width: '120px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '接触职业病危害因素',
    dataIndex: 'hazardFactors',
    width: '180px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '防护措施',
    dataIndex: 'safeguardProcedures',
    width: '120px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '180px',
    align: 'center',
    fixed: 'right',
    scopedSlots: { customRender: 'action' }
  }
]

export const anamnesisColumns = [
  {
    title: '疾病名称',
    dataIndex: 'diseasesName',
    width: '200px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '诊断时间',
    dataIndex: 'diagnosisTime',
    width: '180px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '诊断医院',
    dataIndex: 'diagnosisHospital',
    width: '180px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '治疗结果',
    dataIndex: 'therapeuticOutcome',
    width: '180px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '备注',
    dataIndex: 'remark',
    // width: '120px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '180px',
    align: 'center',
    fixed: 'right',
    scopedSlots: { customRender: 'action' }
  }
]

export const diseaseDiagnosisColumns = [
  {
    title: '疾病名称',
    dataIndex: 'diseaseName',
    width: '200px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '诊断时间',
    dataIndex: 'diagnosisTime',
    width: '180px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '诊断医院',
    dataIndex: 'diagnosisHospital',
    width: '180px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '诊断级别',
    dataIndex: 'diagnosisLevel',
    width: '180px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '备注',
    dataIndex: 'remark',
    // width: '120px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '180px',
    align: 'center',
    // fixed: 'right',
    scopedSlots: { customRender: 'action' }
  }
]

export const labourColumns = [
  {
    title: '序号',
    dataIndex: 'number',
    width: '60px',
    align: 'center',
    scopedSlots: { customRender: 'number' }
  },
  {
    title: '单据编号',
    dataIndex: 'serialNumber',
    width: '200px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '车间',
    dataIndex: 'workshopName',
    width: '180px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '领用人',
    dataIndex: 'recipientName',
    width: '180px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '领用日期',
    dataIndex: 'acceptTime',
    width: '180px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '管理员',
    dataIndex: 'createName',
    width: '120px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '创建日期',
    dataIndex: 'createTime',
    width: '120px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '状态',
    dataIndex: 'status',
    width: '100px',
    align: 'center',
    scopedSlots: { customRender: 'status' }
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '180px',
    align: 'center',
    // fixed: 'right',
    scopedSlots: { customRender: 'action' }
  }
]

export const materialInfoColumns = [
  {
    title: '序号',
    dataIndex: 'number',
    width: '60px',
    align: 'center',
    scopedSlots: { customRender: 'number' }
  },
  {
    title: '代码',
    dataIndex: 'number',
    width: '200px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '物资名称',
    dataIndex: 'name',
    width: '180px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '计量单位',
    dataIndex: 'measurementUnit',
    width: '180px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '备注',
    dataIndex: 'remark',
    width: '120px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '180px',
    align: 'center',
    // fixed: 'right',
    scopedSlots: { customRender: 'action' }
  }
]

export const labourMaterialColumns = [
  {
    title: '序号',
    customRender: (text, record, index) => `${index + 1}`,
    dataIndex: 'number',
    width: '60px',
    align: 'center'
  },
  {
    title: '护品名称',
    dataIndex: 'name',
    width: '180px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '数量',
    dataIndex: 'number',
    width: '140px',
    align: 'center',
    scopedSlots: { customRender: 'number' }
  },
  {
    title: '单位',
    dataIndex: 'unit',
    width: '120px',
    align: 'center',
    ellipsis: true
  },
  {
    title: '操作',
    dataIndex: 'action',
    width: '180px',
    align: 'center',
    fixed: 'right',
    scopedSlots: { customRender: 'action' }
  }
]
