// spang-web/src/config/router.config.js
// eslint-disable-next-line
import { UserLayout, BasicLayout, BlankLayout } from '@/layouts'

// const RouteView = {
//   name: 'RouteView',
//   render: (h) => h('router-view')
// }

export const asyncRouterMap = [

]

/**
 * 基础路由
 * @type { *[] }
 */
export const constantRouterMap = [
  {
    path: '/user',
    component: UserLayout,
    redirect: '/user/login',
    hidden: true,
    children: [
      {
        path: 'login',
        name: 'login',
        component: () => import(/* webpackChunkName: "user" */ '@/views/user/Login')
      }
      // {
      //   path: 'register',
      //   name: 'register',
      //   component: () => import(/* webpackChunkName: "user" */ '@/views/user/Register')
      // },
      // {
      //   path: 'register-result',
      //   name: 'registerResult',
      //   component: () => import(/* webpackChunkName: "user" */ '@/views/user/RegisterResult')
      // },
      // {
      //   path: 'recover',
      //   name: 'recover',
      //   component: undefined
      // }
    ]
  },
  {
    path: '/',
    name: 'index',
    redirect: '/safety',
    component: undefined
  },
  // {
  //   path: '/system',
  //   name: 'system',
  //   component: BasicLayout,
  //   redirect: '/system/workplace',
  //   hidden: true,
  //   meta: { title: '智能管理平台' },
  //   children: [
  //     {
  //       path: '/system/workplace',
  //       name: 'workplace',
  //       component: () => import('@/views/system/index1'),
  //       meta: { title: '工作台', icon: 'appstore', hiddenHeaderContent: true }
  //     },
  //     {
  //       path: '/system/index1',
  //       name: 'index1',
  //       component: () => import('@/views/system/index'),
  //       meta: { title: '菜单一', icon: 'menu' }
  //     },
  //     {
  //       path: '/system/index2',
  //       name: 'index2',
  //       component: undefined,
  //       meta: { title: '菜单二', icon: 'menu' }
  //     },
  //     {
  //       path: '/system/index3',
  //       name: 'index3',
  //       component: undefined,
  //       meta: { title: '菜单三', icon: 'menu' }
  //     },
  //     {
  //       path: '/system/index4',
  //       name: 'index4',
  //       component: undefined,
  //       meta: { title: '菜单四', icon: 'menu' }
  //     }
  //   ]
  // },
  {
    path: '/blank',
    component: BasicLayout
  },
  {
    path: '/404',
    component: () => import(/* webpackChunkName: "fail" */ '@/views/exception/404')
  }
  // {
  //   path: '*', redirect: '/404', hidden: true
  // }

]
